<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.4">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>4&nbsp; Expression quantitative trait locus (eQTL) analysis â€“ Master of Science in Omics Data Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./biblio.html" rel="next">
<link href="./wes.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./wat&amp;msld.html">Supplementary Material</a></li><li class="breadcrumb-item"><a href="./eqtl.html"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Expression quantitative trait locus (eQTL) analysis</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Master of Science in Omics Data Analysis</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/arbones/TFM" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Front Page</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>Manuscript</strong></span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Supplementary Material</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./wat&amp;msld.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Adipose tissue and MASLD</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./rnaseq.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Bulk RNA sequencing</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./wes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Whole exome sequencing</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./eqtl.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Expression quantitative trait locus (eQTL) analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./biblio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Bibliography</span></span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="header-section-number">4.1</span> Introduction</a></li>
  <li><a href="#methods" id="toc-methods" class="nav-link" data-scroll-target="#methods"><span class="header-section-number">4.2</span> Methods</a></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results"><span class="header-section-number">4.3</span> Results</a>
  <ul class="collapse">
  <li><a href="#analysis-of-eqtl" id="toc-analysis-of-eqtl" class="nav-link" data-scroll-target="#analysis-of-eqtl"><span class="header-section-number">4.3.1</span> Analysis of eQTL</a></li>
  <li><a href="#steatosis-eqtl" id="toc-steatosis-eqtl" class="nav-link" data-scroll-target="#steatosis-eqtl"><span class="header-section-number">4.3.2</span> Steatosis-eQTL</a></li>
  <li><a href="#non-steatosis-eqtl" id="toc-non-steatosis-eqtl" class="nav-link" data-scroll-target="#non-steatosis-eqtl"><span class="header-section-number">4.3.3</span> Non steatosis-eQTL</a></li>
  <li><a href="#identification-of-steatosis-specific-eqtls-and-their-association-with-steatosis-phenotypes" id="toc-identification-of-steatosis-specific-eqtls-and-their-association-with-steatosis-phenotypes" class="nav-link" data-scroll-target="#identification-of-steatosis-specific-eqtls-and-their-association-with-steatosis-phenotypes"><span class="header-section-number">4.3.4</span> Identification of Steatosis-Specific eQTLs and Their Association with Steatosis Phenotypes</a></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion"><span class="header-section-number">4.3.5</span> Conclusion</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./wat&amp;msld.html">Supplementary Material</a></li><li class="breadcrumb-item"><a href="./eqtl.html"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Expression quantitative trait locus (eQTL) analysis</span></a></li></ol></nav>
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Expression quantitative trait locus (eQTL) analysis</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(compareGroups)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SNPRelate)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MatrixEQTL)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SNPassoc)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>conflicted<span class="sc">::</span><span class="fu">conflicts_prefer</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(), </span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">filter</span>(),</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>                             base<span class="sc">::</span><span class="fu">setdiff</span>(),</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>                             base<span class="sc">::</span><span class="fu">rownames</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="introduction" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="introduction"><span class="header-section-number">4.1</span> Introduction</h2>
<p>Genome-wide association studies (GWAS) have identified loci associated with NAFLD, but these findings often have modest effect sizes and explain only a small portion of the heritability of NAFLD, which is estimated to be between 22â€“50% <span class="citation" data-cites="anstee2013">(<a href="biblio.html#ref-anstee2013" role="doc-biblioref">Anstee et al., 2013</a>)</span>. While polygenic risk scores (PRS) have shown promise in predicting risk for complex traits <span class="citation" data-cites="trÃ©po2020">(<a href="biblio.html#ref-trÃ©po2020" role="doc-biblioref">TrÃ©po et al., 2020</a>)</span>, their application to NAFLD has yet to yield robust results <span class="citation" data-cites="kawaguchi2018">(<a href="biblio.html#ref-kawaguchi2018" role="doc-biblioref">Kawaguchi et al., 2018</a>)</span>.</p>
<p>To enhance our understanding of the genetic basis of NAFLD, mapping expression quantitative trait loci (eQTLs) provides a powerful approach. eQTL analysis links specific genetic variants with changes in gene expression, offering interpretable molecular connections between genetic variants and traits of interest <span class="citation" data-cites="albert2015 yoo2021">(<a href="biblio.html#ref-albert2015" role="doc-biblioref">Albert et al., 2015</a>; <a href="biblio.html#ref-yoo2021" role="doc-biblioref">Yoo et al., 2021</a>)</span>. This method significantly increases statistical power, allowing the detection of thousands of eQTLs even in relatively small sample sizes <span class="citation" data-cites="montgomery2010 yoo2021">(<a href="biblio.html#ref-montgomery2010" role="doc-biblioref">Montgomery et al., 2010</a>; <a href="biblio.html#ref-yoo2021" role="doc-biblioref">Yoo et al., 2021</a>)</span>.</p>
<p>In the context of obesity, where both adipose tissue and liver function are often compromised, integrating eQTL analysis within a multi-omic framework is essential. The goal of this study is to identify and characterize disease-specific eQTLs in patients with steatosis using gene expression in the liver and whole exome sequencing data. We aim to uncover novel genetic mechanisms underlying NAFLD and to improve the accuracy of genetic risk prediction for this complex disease.</p>
</section>
<section id="methods" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="methods"><span class="header-section-number">4.2</span> Methods</h2>
<p>Transcriptome and genome data were obtained as detailed in Appendices 2 and 3, respectively. No genotype imputation was performed. These datasets were integrated for expression quantitative trait loci (eQTL) mapping using R and <code>MatrixEQTL</code> (version 2.3) <span class="citation" data-cites="MatrixEQTL">(<a href="biblio.html#ref-MatrixEQTL" role="doc-biblioref">Shabalin, 2012</a>)</span>, with sex included as a covariate.</p>
<p>To identify steatosis-related eQTLs, the cohort was divided into two groups: no-steatosis (liver fat content &lt; 5%, n = 29) and steatosis (liver fat content &gt; 33%, n = 25). eQTL mapping was conducted separately within each group. eQTLs identified in both groups were excluded from further analysis to focus on steatosis-specific eQTLs.</p>
<p>To refine the list of steatosis-specific eQTLs, we selected the most significant SNP per gene, retaining only those eQTLs with an absolute Î² coefficient greater than 10. These eQTLs were further filtered by removing any SNPs that did not meet Hardy-Weinberg equilibrium (HWE) criteria (HWE p-value &lt; 0.0001).</p>
<p>A genome-wide association analysis was then performed between the identified SNPs and the dependent variable (steatosis) in the whole cohort using five different genetic models (inheritance patterns): codominant, dominant, recessive, overdominant, and log-additive, utilizing the R package <code>SNPassoc</code> (version 2.1.0) <span class="citation" data-cites="SNPassoc">(<a href="biblio.html#ref-SNPassoc" role="doc-biblioref">Moreno et al., 2022</a>)</span>. For each eQTL, the model with the lowest p-value was selected for further analysis. Finally, the results were visualized by plotting the association data across three histologically defined steatosis groups (&lt;5%, 5-33%, and &gt;33%). Genotypes were coded as 0, 1, or 2, corresponding to the number of minor alleles present at each SNP.</p>
</section>
<section id="results" class="level2" data-number="4.3">
<h2 data-number="4.3" class="anchored" data-anchor-id="results"><span class="header-section-number">4.3</span> Results</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">###################</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Genotype data </span><span class="al">###</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="do">###################</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert PLINK to GDS format</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">snpgdsBED2GDS</span>(<span class="st">"data/only_snps_cleaned.bed"</span>, </span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>              <span class="st">"data/only_snps_cleaned.fam"</span>,</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>              <span class="st">"data/only_snps_cleaned.bim"</span>, </span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">out.gdsfn =</span><span class="st">"genotype.gds"</span>)</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Open the GDS file</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>genofile <span class="ot">&lt;-</span> <span class="fu">snpgdsOpen</span>(<span class="st">"genotype.gds"</span>)</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a><span class="do">###################</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a><span class="co"># phenotype data ##</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a><span class="do">###################</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"data/20240813_metadata.csv"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(sample, <span class="at">.keep_all =</span> T) <span class="sc">|&gt;</span> </span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">paste0</span>(<span class="st">"DNA_"</span>,sample,<span class="st">"_"</span>,<span class="st">"DNA_"</span>,sample))<span class="sc">|&gt;</span> </span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># recode steatosis categories into a new variable with only 3 levels</span></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>     <span class="fu">mutate</span>(<span class="at">steatosis_3cat=</span><span class="fu">fct_recode</span>(Steatosis,<span class="st">`</span><span class="at">&gt;33%</span><span class="st">`</span><span class="ot">=</span><span class="st">"&gt;33-66%"</span>,<span class="st">`</span><span class="at">&gt;33%</span><span class="st">`</span><span class="ot">=</span><span class="st">"&gt;66%"</span>),</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">steatosis_3cat=</span><span class="fu">factor</span>(steatosis_3cat,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">"&lt;5%"</span>, <span class="st">"5-33%"</span>, <span class="st">"&gt;33%"</span>))) <span class="sc">|&gt;</span> </span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>    <span class="co"># recode sex as the .fam file </span></span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">sex =</span> <span class="fu">fct_recode</span>(sexo, <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"hombre"</span>,  <span class="st">"2"</span> <span class="ot">=</span><span class="st">"mujer"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">column_to_rownames</span>(<span class="st">"sample"</span>) </span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>steato <span class="ot">&lt;-</span> pheno <span class="sc">|&gt;</span> <span class="fu">filter</span>(steatosis_3cat <span class="sc">==</span> <span class="st">"&gt;33%"</span>) <span class="sc">|&gt;</span> <span class="fu">pull</span>(prefix)</span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>healthy <span class="ot">&lt;-</span> pheno <span class="sc">|&gt;</span> <span class="fu">filter</span>(steatosis_3cat <span class="sc">==</span> <span class="st">"&lt;5%"</span>) <span class="sc">|&gt;</span> <span class="fu">pull</span>(prefix)</span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################</span></span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a><span class="co"># RNAseq count data #######################</span></span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a><span class="do">###########################################</span></span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>counts0 <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"data/salmon.merged.gene_counts.tsv"</span>)</span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>filter <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(counts0 <span class="sc">&gt;=</span> <span class="dv">30</span>) <span class="sc">&gt;=</span> <span class="dv">30</span></span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a>counts_filtered <span class="ot">&lt;-</span> counts0[filter,] <span class="sc">|&gt;</span> </span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">round</span>(<span class="at">digits =</span> <span class="dv">0</span>)</span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a><span class="do">#########################################################</span></span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a><span class="co"># Map ENSEMBL gene IDs to HGNC symbols for compatibility with gene annotations</span></span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a>symb <span class="ot">&lt;-</span> <span class="fu">mapIds</span>(org.Hs.eg.db, <span class="at">keys =</span> <span class="fu">str_sub</span>(<span class="fu">rownames</span>(counts_filtered),<span class="dv">1</span>,<span class="dv">15</span>),</span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a>               <span class="at">column =</span> <span class="fu">c</span>(<span class="st">"SYMBOL"</span>), <span class="at">keytype =</span> <span class="st">"ENSEMBL"</span>)</span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a><span class="co"># filter out untranslated genes</span></span>
<span id="cb2-49"><a href="#cb2-49" aria-hidden="true" tabindex="-1"></a>counts_na <span class="ot">&lt;-</span> counts_filtered <span class="sc">|&gt;</span> </span>
<span id="cb2-50"><a href="#cb2-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">symbol =</span> symb, <span class="at">.before =</span> <span class="dv">1</span>) <span class="sc">|&gt;</span> </span>
<span id="cb2-51"><a href="#cb2-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(symbol)) </span>
<span id="cb2-52"><a href="#cb2-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-53"><a href="#cb2-53" aria-hidden="true" tabindex="-1"></a><span class="co"># sum the counts of duplicated genes (normaly pseudogenes)</span></span>
<span id="cb2-54"><a href="#cb2-54" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> counts_na <span class="sc">|&gt;</span> </span>
<span id="cb2-55"><a href="#cb2-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(symbol) <span class="sc">|&gt;</span> </span>
<span id="cb2-56"><a href="#cb2-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), \(x) <span class="fu">sum</span>(x))) <span class="sc">|&gt;</span> </span>
<span id="cb2-57"><a href="#cb2-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(symbol,<span class="at">.keep_all =</span> T) <span class="sc">|&gt;</span> </span>
<span id="cb2-58"><a href="#cb2-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">|&gt;</span> </span>
<span id="cb2-59"><a href="#cb2-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"symbol"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="analysis-of-eqtl" class="level3" data-number="4.3.1">
<h3 data-number="4.3.1" class="anchored" data-anchor-id="analysis-of-eqtl"><span class="header-section-number">4.3.1</span> Analysis of eQTL</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>expression_liver <span class="ot">&lt;-</span> counts <span class="sc">|&gt;</span> </span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">"L_"</span>)) </span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>sample_id_liver <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"DNA_"</span>,<span class="fu">str_sub</span>(<span class="fu">names</span>(expression_liver),<span class="dv">3</span>,<span class="dv">5</span>),<span class="st">"_"</span>,<span class="st">"DNA_"</span>,<span class="fu">str_sub</span>(<span class="fu">names</span>(expression_liver),<span class="dv">3</span>,<span class="dv">5</span>))</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract genotype data from GDS file</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>geno_data <span class="ot">&lt;-</span> <span class="fu">snpgdsGetGeno</span>(genofile, <span class="at">sample.id=</span>sample_id_liver, <span class="at">snp.id=</span><span class="cn">NULL</span>, <span class="at">with.id=</span><span class="cn">TRUE</span>)</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the genotype matrix</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>geno_matrix <span class="ot">&lt;-</span> geno_data<span class="sc">$</span>genotype</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Assign row and column names correctly</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(geno_matrix) <span class="ot">&lt;-</span> geno_data<span class="sc">$</span>sample.id  <span class="co"># Row names: sample IDs</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(geno_matrix) <span class="ot">&lt;-</span> geno_data<span class="sc">$</span>snp.id     <span class="co"># Column names: SNP IDs</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder geno_matrix pheno with the same order than the expression matrix</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>geno_matrix_liver <span class="ot">&lt;-</span> geno_matrix[sample_id_liver,]</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>pheno_liver <span class="ot">&lt;-</span> pheno[sample_id_liver,]</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(expression_liver) <span class="ot">&lt;-</span> sample_id_liver</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define model (linear model with covariates, including treatment)</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>model <span class="ot">&lt;-</span> modelLINEAR_CROSS</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare SNP, gene, and covariate data for MatrixEQTL</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>SNP <span class="ot">&lt;-</span> SlicedData<span class="sc">$</span><span class="fu">new</span>()</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>SNP<span class="sc">$</span><span class="fu">CreateFromMatrix</span>(<span class="fu">as.matrix</span>(<span class="fu">t</span>(geno_matrix_liver)))</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>gene <span class="ot">&lt;-</span> SlicedData<span class="sc">$</span><span class="fu">new</span>()</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>gene<span class="sc">$</span><span class="fu">CreateFromMatrix</span>(<span class="fu">as.matrix</span>(expression_liver))</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>cvrt <span class="ot">&lt;-</span> SlicedData<span class="sc">$</span><span class="fu">new</span>()</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>cvrt<span class="sc">$</span><span class="fu">CreateFromMatrix</span>(<span class="fu">as.matrix</span>(<span class="fu">t</span>(<span class="fu">as.numeric</span>(pheno_liver<span class="sc">$</span>sex))))</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Run the eQTL analysis</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>eqtl_results <span class="ot">&lt;-</span> <span class="fu">Matrix_eQTL_main</span>(</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">snps =</span> SNP,</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene =</span> gene,</span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">cvrt =</span> cvrt,</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">output_file_name =</span> <span class="st">"eqtl_results_liver.txt"</span>,</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">pvOutputThreshold =</span> <span class="fl">1e-5</span>,</span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">useModel =</span> model,</span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">pvalue.hist =</span> <span class="cn">TRUE</span>,</span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pv.by.genesnp =</span> <span class="cn">FALSE</span></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a><span class="do">### The model is saved as it need a lot of memory to be executed</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the saved eQTL results from the liver data file</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>eqtl_liver <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"data/eqtl_results_liver.txt"</span>, <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">escape_double =</span> <span class="cn">FALSE</span>, <span class="at">trim_ws =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the false discovery rate (FDR) threshold and the absolute beta coefficient threshold</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>fdr <span class="ot">&lt;-</span> <span class="fl">0.001</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter the eQTL results to include only those with an FDR below the threshold </span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co"># and an absolute beta coefficient above the threshold</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>eqtl_top_liver <span class="ot">&lt;-</span> eqtl_liver <span class="sc">|&gt;</span> </span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> fdr, <span class="fu">abs</span>(beta) <span class="sc">&gt;</span> b )</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Group by SNP and count the number of distinct genes associated with each SNP</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>t <span class="ot">&lt;-</span> eqtl_top_liver <span class="sc">|&gt;</span> </span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SNP) <span class="sc">|&gt;</span> </span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">n_distinct</span>(gene))</span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the percentage of SNPs that affect multiple genes</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>multiple <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">sum</span>(t<span class="sc">$</span>n <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">/</span> <span class="fu">nrow</span>(t) <span class="sc">*</span> <span class="dv">100</span>, <span class="at">digits =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>After quality assessment of the genotype and liver transcriptome data from 79 individuals with obesity , 14564 transcripts and 48852 SNPs were subject eQTL mapping using an additive linear model. We identified 1244 genes with eQTLs (cis and trans) at FDR = 0.001, and 4031 significant SNPs. Among the eQTL-SNPs, 57.7% had an effect on multiple genes, , highlighting the potential pleiotropy of these genetic variants in the regulation of gene expression.</p>
<p>This approach allowed us to uncover a significant number of eQTLs with robust statistical support, offering insights into the genetic regulation of liver gene expression in individuals with obesity. The high percentage of SNPs affecting multiple genes suggests complex regulatory mechanisms at play, which may contribute to the pathogenesis of related diseases.</p>
</section>
<section id="steatosis-eqtl" class="level3" data-number="4.3.2">
<h3 data-number="4.3.2" class="anchored" data-anchor-id="steatosis-eqtl"><span class="header-section-number">4.3.2</span> Steatosis-eQTL</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract liver RNAseq data from individuals with steatosis</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>expression_steato <span class="ot">&lt;-</span> counts  <span class="sc">|&gt;</span> </span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">all_of</span>(steato))</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>sample_id_steato <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"DNA_"</span>,<span class="fu">str_sub</span>(<span class="fu">names</span>(expression_steato),<span class="dv">3</span>,<span class="dv">5</span>),<span class="st">"_"</span>,<span class="st">"DNA_"</span>,<span class="fu">str_sub</span>(<span class="fu">names</span>(expression_steato),<span class="dv">3</span>,<span class="dv">5</span>))</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract genotype data from GDS file</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>geno_data <span class="ot">&lt;-</span> <span class="fu">snpgdsGetGeno</span>(genofile, <span class="at">sample.id=</span>sample_id_steato, <span class="at">snp.id=</span><span class="cn">NULL</span>, <span class="at">with.id=</span><span class="cn">TRUE</span>)</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the genotype matrix</span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>geno_matrix <span class="ot">&lt;-</span> geno_data<span class="sc">$</span>genotype</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Assign row and column names correctly</span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(geno_matrix) <span class="ot">&lt;-</span> geno_data<span class="sc">$</span>sample.id  <span class="co"># Row names: sample IDs</span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(geno_matrix) <span class="ot">&lt;-</span> geno_data<span class="sc">$</span>snp.id     <span class="co"># Column names: SNP IDs</span></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder geno_matrix pheno with the same order than the expression matrix</span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>geno_matrix <span class="ot">&lt;-</span> geno_matrix[sample_id_steato,]</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>pheno_steato <span class="ot">&lt;-</span> pheno[sample_id_steato,]</span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(expression_steato) <span class="ot">&lt;-</span> sample_id_steato</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define model (linear model with covariates, including treatment)</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>model <span class="ot">&lt;-</span> modelLINEAR_CROSS</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare SNP, gene, and covariate data for MatrixEQTL</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>SNP <span class="ot">&lt;-</span> SlicedData<span class="sc">$</span><span class="fu">new</span>()</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>SNP<span class="sc">$</span><span class="fu">CreateFromMatrix</span>(<span class="fu">as.matrix</span>(<span class="fu">t</span>(geno_matrix)))</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>gene <span class="ot">&lt;-</span> SlicedData<span class="sc">$</span><span class="fu">new</span>()</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>gene<span class="sc">$</span><span class="fu">CreateFromMatrix</span>(<span class="fu">as.matrix</span>(expression_steato))</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>cvrt <span class="ot">&lt;-</span> SlicedData<span class="sc">$</span><span class="fu">new</span>()</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>cvrt<span class="sc">$</span><span class="fu">CreateFromMatrix</span>(<span class="fu">as.matrix</span>(<span class="fu">t</span>(<span class="fu">as.numeric</span>(pheno_steato<span class="sc">$</span>sex))))</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Run the eQTL analysis</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>eqtl_results <span class="ot">&lt;-</span> <span class="fu">Matrix_eQTL_main</span>(</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">snps =</span> SNP,</span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene =</span> gene,</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">cvrt =</span> cvrt,</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">output_file_name =</span> <span class="st">"eqtl_results_steato.txt"</span>,</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">pvOutputThreshold =</span> <span class="fl">1e-5</span>,</span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">useModel =</span> model,</span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">pvalue.hist =</span> <span class="cn">TRUE</span>,</span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pv.by.genesnp =</span> <span class="cn">FALSE</span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the results</span></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a><span class="co"># summary(eqtl_results)</span></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a><span class="co"># plot(eqtl_results)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the saved eQTL results from the steatosis data file</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>eqtl_steato <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"data/eqtl_results_steato.txt"</span>, <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">escape_double =</span> <span class="cn">FALSE</span>, <span class="at">trim_ws =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter the eQTL results to include only those with an FDR below the threshold and an absolute beta coefficient above the threshold</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>eqtl_top_steato <span class="ot">&lt;-</span> eqtl_steato <span class="sc">|&gt;</span> </span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> fdr, <span class="fu">abs</span>(beta) <span class="sc">&gt;</span> b )</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Group by SNP and count the number of distinct genes associated with each SNP</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>t <span class="ot">&lt;-</span> eqtl_top_steato <span class="sc">|&gt;</span> </span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SNP) <span class="sc">|&gt;</span> </span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">n_distinct</span>(gene))</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the percentage of SNPs that affect multiple genes</span></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>multiple <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">sum</span>(t<span class="sc">$</span>n <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">/</span> <span class="fu">nrow</span>(t) <span class="sc">*</span> <span class="dv">100</span>, <span class="at">digits =</span> <span class="dv">1</span>)</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Select the most significant SNP per gene, keeping only the SNP with the lowest FDR and highest absolute beta coefficient</span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>eqtl_single_gene_steato <span class="ot">&lt;-</span> eqtl_top_steato <span class="sc">|&gt;</span> </span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SNP) <span class="sc">|&gt;</span> </span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(FDR, <span class="fu">desc</span>(beta)) <span class="sc">|&gt;</span> </span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">|&gt;</span> </span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(gene, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>In this subgroup of 25 individuals with obesity and statosis , 14564 transcripts and 48852 SNPs were subject eQTL mapping. We identified 1191 genes with eQTLs (cis and trans) at FDR = 0.001, and 18220 significant SNPs. Among the eQTL-SNPs, 70.2% had an effect on multiple genes. This subset of eQTLs provides valuable insights into the specific genetic variants that may contribute to the development and progression of liver steatosis in obese individuals. The high proportion of SNPs affecting multiple genes underscores the complexity of the genetic regulation of liver gene expression in the context of steatosis.</p>
</section>
<section id="non-steatosis-eqtl" class="level3" data-number="4.3.3">
<h3 data-number="4.3.3" class="anchored" data-anchor-id="non-steatosis-eqtl"><span class="header-section-number">4.3.3</span> Non steatosis-eQTL</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select RNAseq data for the non-steatosis subset</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>expression_healthy <span class="ot">&lt;-</span> counts  <span class="sc">|&gt;</span> </span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">all_of</span>(healthy))</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>sample_id_healthy <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"DNA_"</span>,<span class="fu">str_sub</span>(<span class="fu">names</span>(expression_healthy),<span class="dv">3</span>,<span class="dv">5</span>),<span class="st">"_"</span>,<span class="st">"DNA_"</span>,<span class="fu">str_sub</span>(<span class="fu">names</span>(expression_healthy),<span class="dv">3</span>,<span class="dv">5</span>))</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract genotype data from GDS file</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>geno_data <span class="ot">&lt;-</span> <span class="fu">snpgdsGetGeno</span>(genofile, <span class="at">sample.id=</span>sample_id_healthy, <span class="at">snp.id=</span><span class="cn">NULL</span>, <span class="at">with.id=</span><span class="cn">TRUE</span>)</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the genotype matrix</span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>geno_matrix <span class="ot">&lt;-</span> geno_data<span class="sc">$</span>genotype</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Assign row and column names correctly</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(geno_matrix) <span class="ot">&lt;-</span> geno_data<span class="sc">$</span>sample.id  <span class="co"># Row names: sample IDs</span></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(geno_matrix) <span class="ot">&lt;-</span> geno_data<span class="sc">$</span>snp.id     <span class="co"># Column names: SNP IDs</span></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder geno_matrix pheno with the same order than the expression matrix</span></span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>geno_matrix <span class="ot">&lt;-</span> geno_matrix[sample_id_healthy,]</span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>pheno_healthy <span class="ot">&lt;-</span> pheno[sample_id_healthy,]</span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(expression_healthy) <span class="ot">&lt;-</span> sample_id_healthy</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define model (linear model with covariates, including treatment)</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>model <span class="ot">&lt;-</span> modelLINEAR_CROSS</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare SNP, gene, and covariate data for MatrixEQTL</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>SNP <span class="ot">&lt;-</span> SlicedData<span class="sc">$</span><span class="fu">new</span>()</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>SNP<span class="sc">$</span><span class="fu">CreateFromMatrix</span>(<span class="fu">as.matrix</span>(<span class="fu">t</span>(geno_matrix)))</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>gene <span class="ot">&lt;-</span> SlicedData<span class="sc">$</span><span class="fu">new</span>()</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>gene<span class="sc">$</span><span class="fu">CreateFromMatrix</span>(<span class="fu">as.matrix</span>(expression_healthy))</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>cvrt <span class="ot">&lt;-</span> SlicedData<span class="sc">$</span><span class="fu">new</span>()</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>cvrt<span class="sc">$</span><span class="fu">CreateFromMatrix</span>(<span class="fu">as.matrix</span>(<span class="fu">t</span>(<span class="fu">as.numeric</span>(pheno_healthy<span class="sc">$</span>sex))))</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Run the eQTL analysis</span></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>eqtl_results <span class="ot">&lt;-</span> <span class="fu">Matrix_eQTL_main</span>(</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">snps =</span> SNP,</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene =</span> gene,</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">cvrt =</span> cvrt,</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">output_file_name =</span> <span class="st">"eqtl_results_healthy.txt"</span>,</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">pvOutputThreshold =</span> <span class="fl">1e-5</span>,</span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">useModel =</span> model,</span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">pvalue.hist =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pv.by.genesnp =</span> <span class="cn">FALSE</span></span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the results</span></span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a><span class="co"># summary(eqtl_results)</span></span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a><span class="co"># plot(eqtl_results)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the eQTL results from the non-steatosis (healthy) data file</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>eqtl_healthy <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"data/eqtl_results_healthy.txt"</span>, <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">escape_double =</span> <span class="cn">FALSE</span>, <span class="at">trim_ws =</span> <span class="cn">TRUE</span>)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter the eQTL results to include only those with an FDR below the threshold and an absolute beta coefficient above the threshold</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>eqtl_top_healthy <span class="ot">&lt;-</span> eqtl_healthy <span class="sc">|&gt;</span> </span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FDR <span class="sc">&lt;</span> fdr, <span class="fu">abs</span>(beta) <span class="sc">&gt;</span> b )</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Group by SNP and count the number of distinct genes associated with each SNP</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>t <span class="ot">&lt;-</span> eqtl_top_healthy <span class="sc">|&gt;</span> </span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SNP) <span class="sc">|&gt;</span> </span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">n_distinct</span>(gene))</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the percentage of SNPs that affect multiple genes</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>multiple <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">sum</span>(t<span class="sc">$</span>n <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">/</span> <span class="fu">nrow</span>(t) <span class="sc">*</span> <span class="dv">100</span>, <span class="at">digits =</span> <span class="dv">1</span>)</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Select the most significant SNP per gene, keeping only the SNP with the lowest FDR and highest absolute beta coefficient</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>eqtl_single_gene_healthy <span class="ot">&lt;-</span> eqtl_top_healthy <span class="sc">|&gt;</span> </span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SNP) <span class="sc">|&gt;</span> </span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(FDR, <span class="fu">desc</span>(beta)) <span class="sc">|&gt;</span> </span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">|&gt;</span> </span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(gene, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>In the non-steatosis subgroup, encompassed with 29 individuals with obesity , 14564 transcripts and 48852 SNPs were subject eQTL mapping. We identified 764 genes with eQTLs (cis and trans) at FDR = 0.001, and 14510 significant SNPs. Among the eQTL-SNPs, 56.2% had an effect on multiple genes.</p>
<p>This analysis allowed us to identify a significant number of eQTLs specific to individuals without steatosis, providing insights into the genetic regulation of liver gene expression in a healthier metabolic context. The lower proportion of SNPs affecting multiple genes compared to the steatosis group suggests potential differences in the genetic architecture underlying gene expression in individuals with and without liver steatosis.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Close the GDS file after analysis to free resources</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="fu">closefn.gds</span>(genotype.gds) <span class="do">### como se llama ???</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="identification-of-steatosis-specific-eqtls-and-their-association-with-steatosis-phenotypes" class="level3" data-number="4.3.4">
<h3 data-number="4.3.4" class="anchored" data-anchor-id="identification-of-steatosis-specific-eqtls-and-their-association-with-steatosis-phenotypes"><span class="header-section-number">4.3.4</span> Identification of Steatosis-Specific eQTLs and Their Association with Steatosis Phenotypes</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify SNPs that are unique to the steatosis group by finding the difference between the steatosis and healthy SNP sets</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>dif <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(eqtl_single_gene_steato<span class="sc">$</span>SNP, eqtl_single_gene_healthy<span class="sc">$</span>SNP)</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a genotype-phenotype matrix for the liver samples, selecting only the SNPs unique to the steatosis group</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>assoc_liver0 <span class="ot">&lt;-</span> geno_matrix_liver <span class="sc">|&gt;</span> </span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span> </span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">all_of</span>(dif)) <span class="sc">|&gt;</span> </span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">steatosis_3cat =</span> pheno_liver<span class="sc">$</span>steatosis_3cat, <span class="co"># Add the steatosis phenotype in three categories</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">steatosis =</span> <span class="fu">ifelse</span>(steatosis_3cat <span class="sc">==</span> <span class="st">"&gt;33%"</span>, <span class="dv">1</span>, <span class="dv">0</span>), <span class="co"># Create a binary steatosis variable (&gt;33% = 1, otherwise = 0)</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">sex =</span> pheno_liver<span class="sc">$</span>sexo, <span class="at">.before =</span> <span class="dv">1</span>) <span class="co"># Include sex as a covariate</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up the SNP data for association analysis, labeling genotypes as 0, 1, 2</span></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>assoc_liver <span class="ot">&lt;-</span> <span class="fu">setupSNP</span>(assoc_liver0, <span class="at">colSNPs =</span> <span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(assoc_liver0), <span class="at">name.genotypes =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter SNPs that meet the Hardy-Weinberg equilibrium criteria (HWE &gt;= 0.0001)</span></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>ok_liver <span class="ot">&lt;-</span> <span class="fu">summary</span>(assoc_liver, <span class="at">print =</span> F) <span class="sc">|&gt;</span> </span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(HWE <span class="sc">&gt;=</span> <span class="fl">0.0001</span>) </span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare the final dataset for association analysis, selecting relevant variables and SNPs</span></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>assoc_liver <span class="ot">&lt;-</span> assoc_liver <span class="sc">|&gt;</span> </span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(steatosis_3cat, steatosis, sex, <span class="fu">rownames</span>(ok_liver))</span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform genome-wide association analysis using multiple genetic models and select the best model for each SNP</span></span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">89</span>)</span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>final0 <span class="ot">&lt;-</span> <span class="fu">WGassociation</span>(steatosis <span class="sc">~</span> sex, assoc_liver[, <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(assoc_liver)], <span class="at">mc.cores =</span> <span class="dv">8</span>)  </span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Process the association results to identify significant SNPs</span></span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a>final <span class="ot">&lt;-</span> final0 <span class="sc">|&gt;</span> </span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span> </span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"snp"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(codominant<span class="sc">:</span><span class="st">`</span><span class="at">log-additive</span><span class="st">`</span>, <span class="at">names_to =</span> <span class="st">"model"</span>, <span class="at">values_to =</span> <span class="st">"pvalue"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(snp) <span class="sc">|&gt;</span> </span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(pvalue, <span class="at">.by_group =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> </span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(snp, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> </span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(pvalue)) <span class="sc">|&gt;</span> </span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fdr =</span> <span class="fu">p.adjust</span>(pvalue, <span class="at">method =</span> <span class="st">"fdr"</span>)) <span class="sc">|&gt;</span> <span class="co"># Adjust p-values for multiple testing using FDR</span></span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(fdr <span class="sc">&lt;</span> <span class="fl">0.01</span>) <span class="sc">|&gt;</span> </span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate_wider_delim</span>(snp, <span class="at">names =</span> <span class="fu">c</span>(<span class="st">"chr"</span>, <span class="st">"BP"</span>), <span class="at">delim =</span> <span class="st">"."</span>, <span class="at">cols_remove =</span> F) <span class="sc">|&gt;</span> </span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">BP =</span> <span class="fu">as.integer</span> (BP),</span>
<span id="cb13-40"><a href="#cb13-40" aria-hidden="true" tabindex="-1"></a>         <span class="at">position =</span> <span class="fu">str_c</span>(<span class="fu">str_replace</span>(chr, <span class="st">"chr"</span>, <span class="st">""</span>), BP, BP, <span class="at">sep =</span> <span class="st">":"</span>), <span class="co"># Format the SNP positions</span></span>
<span id="cb13-41"><a href="#cb13-41" aria-hidden="true" tabindex="-1"></a>         <span class="at">snp =</span> <span class="fu">str_replace</span>(snp, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">:"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>To identify steatosis-specific eQTLs, we first determined the SNPs that were unique to the steatosis group by comparing eQTLs identified in individuals with and without steatosis. This analysis revealed a set of 760 SNPs specific to the steatosis group.</p>
<p>Next, we performed a genome-wide association analysis on these steatosis-specific SNPs using genotype data from the liver samples. The analysis included a binary steatosis phenotype, where individuals with liver fat content greater than 33% were classified as having steatosis. The association analysis was conducted using multiple genetic models (codominant, dominant, recessive, overdominant, and log-additive), and for each SNP, the model with the lowest p-value was selected.</p>
<p>To ensure robustness, only SNPs that passed the Hardy-Weinberg equilibrium filter (HWE â‰¥ 0.0001) were included in the final analysis. After adjusting for multiple testing using the false discovery rate (FDR) method, we identified 5 significant SNPs associated with the steatosis phenotype at an FDR threshold of 0.01. These results highlight specific genetic variants that may play a critical role in the development and progression of steatosis, offering potential targets for further investigation in the context of non-alcoholic fatty liver disease (NAFLD).</p>
<p>These SNPs were further annotated using biomaRt to map them to corresponding genes and reference SNP IDs. The final list of steatosis-specific eQTLs provided insights into the genetic mechanisms underlying steatosis in the liver, revealing potential candidate genes for further study.</p>
<p>In the analysis of eQTLs associated with liver steatosis, we observed significant differences in allele distribution across the three steatosis categories (&lt;5%, 5-33%, and &gt;33%) (<a href="#tbl-qtl" class="quarto-xref">Table&nbsp;<span>4.1</span></a>). The alleles are coded as 0 for the major allele and 1 for the minor allele; no homozygous (2) minor allele carriers were found in our cohort, likely due to the small sample size.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use biomaRt to retrieve additional SNP information, including gene associations</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>ensembl <span class="ot">&lt;-</span> biomaRt<span class="sc">::</span><span class="fu">useEnsembl</span>(<span class="at">biomart =</span> <span class="st">"snps"</span>, <span class="at">dataset =</span> <span class="st">"hsapiens_snp"</span>)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>snp_mart <span class="ot">&lt;-</span> biomaRt<span class="sc">::</span><span class="fu">getBM</span>(</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">'chrom_start'</span>,<span class="st">'refsnp_id'</span>,<span class="st">"ensembl_gene_stable_id"</span>, <span class="st">'chr_name'</span>, <span class="st">'allele'</span>, <span class="st">"allele_1"</span>),</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">filters =</span> <span class="st">'chromosomal_region'</span>, </span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">values =</span> final<span class="sc">$</span>position ,</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">mart =</span> ensembl) <span class="sc">|&gt;</span> </span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(chrom_start, <span class="at">.keep_all =</span> T)</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Map Ensembl gene IDs to gene symbols</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>gene_symbol <span class="ot">=</span> <span class="fu">mapIds</span>(org.Hs.eg.db, <span class="at">keys =</span> snp_mart<span class="sc">$</span>ensembl_gene_stable_id,</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">column =</span> <span class="fu">c</span>(<span class="st">"SYMBOL"</span>), <span class="at">keytype =</span> <span class="st">"ENSEMBL"</span>)</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Integrate SNP annotation with association results, including reference SNP IDs</span></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>final <span class="ot">&lt;-</span> final <span class="sc">|&gt;</span> </span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(snp_mart, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"BP"</span> <span class="ot">=</span> <span class="st">"chrom_start"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">rfsnp =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(refsnp_id), snp, refsnp_id))</span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare the eQTL data for visualization and comparison across steatosis groups</span></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>eqtl <span class="ot">&lt;-</span> assoc_liver0 <span class="sc">|&gt;</span> </span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(steatosis, steatosis_3cat, sex, final<span class="sc">$</span>snp) <span class="sc">|&gt;</span> </span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), \(x) <span class="fu">as.character</span>(x))) <span class="sc">|&gt;</span> </span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_names</span>(<span class="fu">c</span>(<span class="st">"steatosis"</span>, <span class="st">"steatosis_3cat"</span>, <span class="st">"sex"</span>, final<span class="sc">$</span>rfsnp))</span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>eqtl_long <span class="ot">&lt;-</span>  eqtl <span class="sc">|&gt;</span>  </span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(eqtl), <span class="at">values_to =</span> <span class="st">"Allele"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Allele))</span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Descriptive analysis and table generation</span></span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>cq <span class="ot">&lt;-</span> <span class="fu">compareGroups</span>(steatosis_3cat <span class="sc">~</span> . <span class="sc">-</span>steatosis <span class="sc">-</span> sex, <span class="at">data =</span> eqtl)</span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>tq <span class="ot">&lt;-</span> <span class="fu">createTable</span>(cq, <span class="at">show.all =</span> T)</span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a><span class="fu">export2md</span>(tq, <span class="at">caption =</span> <span class="st">""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="tbl-qtl" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-qtl-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;4.1: Distribution (%) of the major (0) and minor (1) associated with steatosis across different steatosis categories.<br>p.overall: Chi square associated-p value for the comparison between steatosis categories.
</figcaption>
<div aria-describedby="tbl-qtl-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<table class="table table-striped table-condensed do-not-create-environment cell caption-top table-sm small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: center;" data-quarto-table-cell-role="th">[ALL]</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">&lt;5%</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">5-33%</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">&gt;33%</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">p.overall</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-style: italic; border-bottom: 1px solid grey;"></td>
<td style="text-align: center; font-style: italic; border-bottom: 1px solid grey;">N=79</td>
<td style="text-align: center; font-style: italic; border-bottom: 1px solid grey;">N=29</td>
<td style="text-align: center; font-style: italic; border-bottom: 1px solid grey;">N=25</td>
<td style="text-align: center; font-style: italic; border-bottom: 1px solid grey;">N=25</td>
<td style="text-align: center; font-style: italic; border-bottom: 1px solid grey;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">rs3127806:</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;">0.020</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">0</td>
<td style="text-align: center;">56 (82.4%)</td>
<td style="text-align: center;">22 (95.7%)</td>
<td style="text-align: center;">20 (87.0%)</td>
<td style="text-align: center;">14 (63.6%)</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">1</td>
<td style="text-align: center;">12 (17.6%)</td>
<td style="text-align: center;">1 (4.35%)</td>
<td style="text-align: center;">3 (13.0%)</td>
<td style="text-align: center;">8 (36.4%)</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">rs326214:</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;">0.009</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">0</td>
<td style="text-align: center;">43 (58.1%)</td>
<td style="text-align: center;">10 (37.0%)</td>
<td style="text-align: center;">14 (60.9%)</td>
<td style="text-align: center;">19 (79.2%)</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">1</td>
<td style="text-align: center;">31 (41.9%)</td>
<td style="text-align: center;">17 (63.0%)</td>
<td style="text-align: center;">9 (39.1%)</td>
<td style="text-align: center;">5 (20.8%)</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">rs937583:</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;">0.029</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">0</td>
<td style="text-align: center;">54 (68.4%)</td>
<td style="text-align: center;">23 (79.3%)</td>
<td style="text-align: center;">19 (76.0%)</td>
<td style="text-align: center;">12 (48.0%)</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">1</td>
<td style="text-align: center;">25 (31.6%)</td>
<td style="text-align: center;">6 (20.7%)</td>
<td style="text-align: center;">6 (24.0%)</td>
<td style="text-align: center;">13 (52.0%)</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">chr3:9473234:</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;">0.024</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">0</td>
<td style="text-align: center;">42 (53.8%)</td>
<td style="text-align: center;">13 (46.4%)</td>
<td style="text-align: center;">10 (40.0%)</td>
<td style="text-align: center;">19 (76.0%)</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">1</td>
<td style="text-align: center;">36 (46.2%)</td>
<td style="text-align: center;">15 (53.6%)</td>
<td style="text-align: center;">15 (60.0%)</td>
<td style="text-align: center;">6 (24.0%)</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">rs3739310:</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;">0.008</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">0</td>
<td style="text-align: center;">45 (62.5%)</td>
<td style="text-align: center;">17 (60.7%)</td>
<td style="text-align: center;">9 (40.9%)</td>
<td style="text-align: center;">19 (86.4%)</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">1</td>
<td style="text-align: center;">27 (37.5%)</td>
<td style="text-align: center;">11 (39.3%)</td>
<td style="text-align: center;">13 (59.1%)</td>
<td style="text-align: center;">3 (13.6%)</td>
<td style="text-align: center;"></td>
</tr>
</tbody>
</table>


</div>
</div>
</figure>
</div>
</div>
<p>For SNP rs3127806 in the <em>PLCL2</em> gene, the frequency of the minor allele (1) increased with higher steatosis, indicating a potential role in cellular signaling related to steatosis (p = 0.020). SNP rs326214 in the <em>GCKR</em> gene, known for its involvement in glucose metabolism, showed a significant association with higher steatosis severity (p = 0.009). Similarly, rs937583 in the <em>SLC16A1</em> gene, linked to monocarboxylate transport, demonstrated an increasing frequency of the minor allele with greater steatosis (p = 0.029). SNP chr3:9473234 in the <em>RHOA</em> gene, which regulates cellular structure and migration, also showed significant allele distribution differences (p = 0.024). Lastly, rs3739310 in the <em>TM6SF2</em> gene, associated with liver fat metabolism, displayed a significant association with steatosis severity (p = 0.008). These findings highlight the influence of metabolic pathways in the genetic predisposition to liver steatosis (<a href="#fig-bar" class="quarto-xref">Figure&nbsp;<span>4.1</span></a>).</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualization: Bar plot showing the distribution of alleles across steatosis categories</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(eqtl_long, <span class="fu">aes</span>(<span class="at">x =</span> steatosis_3cat, <span class="at">fill =</span> Allele)) <span class="sc">+</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">position =</span> <span class="st">"fill"</span>) <span class="sc">+</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">""</span>, <span class="at">labels =</span> scales<span class="sc">::</span><span class="fu">percent_format</span>()) <span class="sc">+</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">Steatosis"</span>)<span class="sc">+</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_void</span>() <span class="sc">+</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>)) <span class="sc">+</span></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>name, <span class="at">ncol =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-bar" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-bar-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="eqtl_files/figure-html/fig-bar-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-bar-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4.1: Barplot with the distribution (%) of the major (0) and minor (1) associated with steatosis across different steatosis caetgories.
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="conclusion" class="level3" data-number="4.3.5">
<h3 data-number="4.3.5" class="anchored" data-anchor-id="conclusion"><span class="header-section-number">4.3.5</span> Conclusion</h3>
<p>In conclusion, our study identified significant expression quantitative trait loci (eQTLs) associated with liver steatosis by leveraging whole exome sequencing and transcriptomic data from individuals with obesity. We observed that a substantial proportion of eQTL-SNPs have pleiotropic effects, influencing multiple genes, which underscores the complex genetic architecture underlying this condition. By distinguishing steatosis-specific eQTLs from those present in non-steatosis individuals, we have highlighted key genetic variants that may contribute to the pathogenesis of liver steatosis. These findings offer valuable insights into the molecular mechanisms of steatosis and provide a foundation for future research aimed at developing targeted therapeutic interventions for non-alcoholic fatty liver disease (NAFLD).</p>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" data-line-spacing="2" role="list" style="display: none">
<div id="ref-albert2015" class="csl-entry" role="listitem">
Albert, F. W., &amp; Kruglyak, L. (2015). The role of regulatory variation in complex traits and disease. <em>Nature Reviews Genetics</em>, <em>16</em>(4), 197â€“212. doi: <a href="https://doi.org/10.1038/nrg3891">10.1038/nrg3891</a>
</div>
<div id="ref-anstee2013" class="csl-entry" role="listitem">
Anstee, Q. M., &amp; Day, C. P. (2013). The genetics of NAFLD. <em>Nature Reviews Gastroenterology &amp; Hepatology</em>, <em>10</em>(11), 645â€“655. doi: <a href="https://doi.org/10.1038/nrgastro.2013.182">10.1038/nrgastro.2013.182</a>
</div>
<div id="ref-kawaguchi2018" class="csl-entry" role="listitem">
Kawaguchi, T., Shima, T., Mizuno, M., Mitsumoto, Y., Umemura, A., Kanbara, Y., Tanaka, S., Sumida, Y., Yasui, K., Takahashi, M., Matsuo, K., Itoh, Y., Tokushige, K., Hashimoto, E., Kiyosawa, K., Kawaguchi, M., Itoh, H., Uto, H., Komorizono, Y., â€¦ Okanoue, T. (2018). Risk estimation model for nonalcoholic fatty liver disease in the Japanese using multiple genetic markers. <em>PLOS ONE</em>, <em>13</em>(1), e0185490. doi: <a href="https://doi.org/10.1371/journal.pone.0185490">10.1371/journal.pone.0185490</a>
</div>
<div id="ref-montgomery2010" class="csl-entry" role="listitem">
Montgomery, S. B., Sammeth, M., Gutierrez-Arcelus, M., Lach, R. P., Ingle, C., Nisbett, J., Guigo, R., &amp; Dermitzakis, E. T. (2010). Transcriptome genetics using second generation sequencing in a Caucasian population. <em>Nature</em>, <em>464</em>(7289), 773â€“777. doi: <a href="https://doi.org/10.1038/nature08903">10.1038/nature08903</a>
</div>
<div id="ref-SNPassoc" class="csl-entry" role="listitem">
Moreno, V., Gonzalez, J. R., &amp; Pelegri, D. (2022). <em>SNPassoc: SNPs-based whole genome association studies</em>. Retrieved from <a href="https://CRAN.R-project.org/package=SNPassoc">https://CRAN.R-project.org/package=SNPassoc</a>
</div>
<div id="ref-MatrixEQTL" class="csl-entry" role="listitem">
Shabalin, A. A. (2012). <em>Matrix <span></span>eQTL<span></span>: Ultra fast <span></span>eQTL<span></span> analysis via large matrix operations</em>. <em>28</em>. Retrieved from <a href="http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/">http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/</a>
</div>
<div id="ref-trÃ©po2020" class="csl-entry" role="listitem">
TrÃ©po, E., &amp; Valenti, L. (2020). Update on NAFLD genetics: From new variants to the clinic. <em>Journal of Hepatology</em>, <em>72</em>(6), 1196â€“1209. doi: <a href="https://doi.org/10.1016/j.jhep.2020.02.020">10.1016/j.jhep.2020.02.020</a>
</div>
<div id="ref-yoo2021" class="csl-entry" role="listitem">
Yoo, T., Joo, S. K., Kim, H. J., Kim, H. Y., Sim, H., Lee, J., Kim, H.-H., Jung, S., Lee, Y., Jamialahmadi, O., Romeo, S., Jeong, W.-I., Hwang, G.-S., Kang, K. W., Kim, J. W., Kim, W., &amp; Choi, M. (2021). Disease-specific eQTL screening reveals an anti-fibrotic effect of AGXT2 in non-alcoholic fatty liver disease. <em>Journal of Hepatology</em>, <em>75</em>(3), 514â€“523. doi: <a href="https://doi.org/10.1016/j.jhep.2021.04.011">10.1016/j.jhep.2021.04.011</a>
</div>
</div>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./wes.html" class="pagination-link" aria-label="Whole exome sequencing">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Whole exome sequencing</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./biblio.html" class="pagination-link" aria-label="Bibliography">
        <span class="nav-page-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Bibliography</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>