<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.4">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Jose Miguel Arbones Mainar">
<meta name="dcterms.date" content="2024-09-03">

<title>Manuscript – Master of Science in Omics Data Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./wat&amp;msld.html" rel="next">
<link href="./index.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./intro.html"><strong>Manuscript</strong></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Master of Science in Omics Data Analysis</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/arbones/TFM" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Front Page</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><strong>Manuscript</strong></span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Supplementary Material</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./wat&amp;msld.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Adipose tissue and MASLD</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./rnaseq.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Bulk RNA sequencing</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./wes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Whole exome sequencing</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./eqtl.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Expression quantitative trait locus (eQTL) analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./biblio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Bibliography</span></span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#titulo" id="toc-titulo" class="nav-link active" data-scroll-target="#titulo">TITULO</a>
  <ul class="collapse">
  <li><a href="#abstract" id="toc-abstract" class="nav-link" data-scroll-target="#abstract">Abstract</a></li>
  <li><a href="#introduction" id="toc-introduction" class="nav-link" data-scroll-target="#introduction">Introduction</a></li>
  <li><a href="#material-and-methods" id="toc-material-and-methods" class="nav-link" data-scroll-target="#material-and-methods">Material and methods</a>
  <ul class="collapse">
  <li><a href="#human-cohort-description" id="toc-human-cohort-description" class="nav-link" data-scroll-target="#human-cohort-description">Human cohort description</a></li>
  <li><a href="#collection-and-characterization-of-adipose-and-liver-biosies" id="toc-collection-and-characterization-of-adipose-and-liver-biosies" class="nav-link" data-scroll-target="#collection-and-characterization-of-adipose-and-liver-biosies">Collection and characterization of adipose and liver biosies</a></li>
  <li><a href="#genomic-dna-isolation" id="toc-genomic-dna-isolation" class="nav-link" data-scroll-target="#genomic-dna-isolation">Genomic DNA isolation</a></li>
  <li><a href="#whole-exome-sequencing" id="toc-whole-exome-sequencing" class="nav-link" data-scroll-target="#whole-exome-sequencing">Whole Exome Sequencing</a></li>
  <li><a href="#rna-isolation" id="toc-rna-isolation" class="nav-link" data-scroll-target="#rna-isolation">RNA isolation</a></li>
  <li><a href="#rna-sequencing" id="toc-rna-sequencing" class="nav-link" data-scroll-target="#rna-sequencing">RNA Sequencing</a></li>
  <li><a href="#statistical-analyses" id="toc-statistical-analyses" class="nav-link" data-scroll-target="#statistical-analyses">Statistical analyses</a></li>
  </ul></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results">Results</a></li>
  <li><a href="#discusion" id="toc-discusion" class="nav-link" data-scroll-target="#discusion">Discusion</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><strong>Manuscript</strong></h1>
</div>


<div class="quarto-title-meta-author">
  <div class="quarto-title-meta-heading">Author</div>
  <div class="quarto-title-meta-heading">Affiliation</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author"><a href="www.adipofat.com">Jose Miguel Arbones Mainar</a> <a href="mailto:jmarbones.iacs@aragon.es" class="quarto-title-author-email"><i class="bi bi-envelope"></i></a> <a href="https://orcid.org/0000-0002-8982-3737" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
  <div class="quarto-title-meta-contents">
        <p class="affiliation">
            Unidad de Investigación Traslacional<br>Hospital Universitario Miguel Servet<br>Instituto Aragonés de Ciencias de la Salud
          </p>
      </div>
  </div>

<div class="quarto-title-meta">

      
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">September 3, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="titulo" class="level1 unnumbered">
<h1 class="unnumbered">TITULO</h1>
<section id="abstract" class="level2">
<h2 class="anchored" data-anchor-id="abstract">Abstract</h2>
</section>
<section id="introduction" class="level2">
<h2 class="anchored" data-anchor-id="introduction">Introduction</h2>
</section>
<section id="material-and-methods" class="level2">
<h2 class="anchored" data-anchor-id="material-and-methods">Material and methods</h2>
<section id="human-cohort-description" class="level3">
<h3 class="anchored" data-anchor-id="human-cohort-description">Human cohort description</h3>
<p>The cohort used for this study was the FATe cohort, a longitudinal cohort of obese patients undergoing bariatric surgery in Miguel Servet University Hospital (HUMS, Zaragoza, Spain), that donated blood samples, subcutaneous and visceral adipose tissue biopsies and eventually liver biopsies. The FATe cohort is a live cohort in which patients have been recruited at different time points since 2013 and undergone different procedures and techniques according to the research needs <span class="citation" data-cites="torres-perez2015">(<a href="biblio.html#ref-torres-perez2015" role="doc-biblioref">Torres-Perez et al., 2015</a>)</span>.</p>
<p>Patients were screened to exclude alcohol or drugs abuse, autoimmune disease, chronic inflammatory diseases or chronic infectious diseases (HIV, HBV, HCV). All subjects in the cohort provided written consent and the study was approved by the Regional Institutional Review Board of Ethics at Aragón, Spain (CEIC-A). Samples from patients included in this study were stored by the Biobank of the Aragon Health System (PT20/00112), integrated in the Spanish National Biobanks Network and they were processed following standard operating procedures.</p>
</section>
<section id="collection-and-characterization-of-adipose-and-liver-biosies" class="level3">
<h3 class="anchored" data-anchor-id="collection-and-characterization-of-adipose-and-liver-biosies">Collection and characterization of adipose and liver biosies</h3>
<p>Biopsies (~3 cm3) of AT from both the subcutaneous and the visceral fat depot were obtained during surgery with a bipolar/ultra-sonic device (Thunderbeat. Olympus, Spain) and extracted <em>via</em> a 12 mm trocar (Applied Medical Europe, Spain) inserted in the left hypochondrium during laparoscopic surgery. Liver biopsies were also obtained during the surgery and were deemed adequate when they were at least 16 mm in length and included at least 6 portal tracts.</p>
<p>Paraffin sections of livers were stained with hematoxylin-eosin (H&amp;E). Pathological features of steatosis, lobular inflammation, hepatocellular ballooning, and fibrosis were scored by an experienced pathologist, according to criteria established by the Nonalcoholic Steatohepatitis Clinical Research Network <span class="citation" data-cites="kleiner2005">(<a href="biblio.html#ref-kleiner2005" role="doc-biblioref">Kleiner et al., 2005</a>)</span>.</p>
</section>
<section id="genomic-dna-isolation" class="level3">
<h3 class="anchored" data-anchor-id="genomic-dna-isolation">Genomic DNA isolation</h3>
<p>DNA from buffy coat was isolated</p>
<p>The concentration and purity of the gDNA were determined by measuring absorbance at 260/280 nm and 260/230 nm using a Nanodrop 2000 (Thermo Fisher).</p>
</section>
<section id="whole-exome-sequencing" class="level3">
<h3 class="anchored" data-anchor-id="whole-exome-sequencing">Whole Exome Sequencing</h3>
<p>Genomic DNA was randomly sheared into fragments of 180-280 bp, which were end-repaired, A-tailed, and ligated with Illumina adapters. The adapter-ligated fragments were PCR amplified, size selected, and purified. Hybridization capture was performed using biotin-labeled probes, with magnetic beads used for exon capture, followed by washing and probe digestion. The enriched libraries were quantified, pooled, and sequenced by Novogene on Illumina platforms.</p>
<p>The computational analysis was managed using Nextflow (v23.10.1) with the <code>nf-core/sarek</code>pipeline (version 3.4.1) <span class="citation" data-cites="garcia2020">(<a href="biblio.html#ref-garcia2020" role="doc-biblioref">Garcia et al., 2020</a>)</span>, optimized for variant calling. Sequencing data were aligned to the GATK GRCh38 human reference genome, and Freebayes was used to call variants <span class="citation" data-cites="garrison2012">(<a href="biblio.html#ref-garrison2012" role="doc-biblioref">Garrison et al., 2012</a>)</span>. The resulting variants were stored in Variant Call Format (VCF) files, which contain detailed metadata and structured information about each detected variant. The process adhered to GATK best practices for quality score recalibration and variant filtering, ensuring accuracy and reliability. Further postprocessing of the obtained VCF files involved normalization using <code>bcftools</code> <span class="citation" data-cites="li2011">(<a href="biblio.html#ref-li2011" role="doc-biblioref">Li, 2011</a>)</span>, merging VCF files, and removing duplicate variants. The final VCF file was transformed into a binary ped format using <code>PLINK</code><span class="citation" data-cites="purcell2007">(<a href="biblio.html#ref-purcell2007" role="doc-biblioref">Purcell et al., 2007</a>)</span>, with filters applied for minor allele frequency, genotyping completeness, and Hardy-Weinberg equilibrium.</p>
</section>
<section id="rna-isolation" class="level3">
<h3 class="anchored" data-anchor-id="rna-isolation">RNA isolation</h3>
<p>Total RNA was isolated from frozen biopsies of scWAT and cell cultures using TRIzol reagent following the manufacturer’s protocol. After homogenization with TRIzol, chloroform was added to the lysates, followed by vigorous shaking, incubation, and centrifugation to separate the phases. The aqueous phase containing RNA was isolated, precipitated with isopropanol, washed with ethanol, and then resuspended in DEPC water. RNA samples were treated with RNase-Free DNase to remove genomic DNA, and their concentration, purity, and quality were assessed using a Nanodrop and agarose gel electrophoresis.</p>
</section>
<section id="rna-sequencing" class="level3">
<h3 class="anchored" data-anchor-id="rna-sequencing">RNA Sequencing</h3>
<p>RNA integrity was assessed using the RNA Integrity Number (RIN) on the Agilent 2200 TapeStation with the RNA ScreenTape assay. Stranded mRNA libraries were prepared using the Novogene NGS RNA Library Prep Set (PT042), involving poly-T oligo-attached magnetic bead mRNA isolation, cDNA synthesis, adapter ligation, and PCR amplification. Libraries passing quality controls were sequenced (2x150 bp) on the Illumina Novaseq X Plus platform. The <code>nf-core/rnaseq</code> pipeline <span class="citation" data-cites="ewels2020">(<a href="biblio.html#ref-ewels2020" role="doc-biblioref">Ewels et al., 2020</a>)</span> in Nextflow (version 23.10.1) was employed for data processing, including trimming with <code>Cutadapt</code> <span class="citation" data-cites="martin2011">(<a href="biblio.html#ref-martin2011" role="doc-biblioref">Martin, 2011</a>)</span>, alignment with <code>STAR</code> <span class="citation" data-cites="dobin2012">(<a href="biblio.html#ref-dobin2012" role="doc-biblioref">Dobin et al., 2012</a>)</span>, and quantification with <code>Salmon</code> <span class="citation" data-cites="patro2017">(<a href="biblio.html#ref-patro2017" role="doc-biblioref">Patro et al., 2017</a>)</span>, using GRCh38.p13 and Gencode v40 as references. Quality control was conducted throughout the process, with mapping efficiencies averaging around 90% for both liver and scWAT samples.</p>
</section>
<section id="statistical-analyses" class="level3">
<h3 class="anchored" data-anchor-id="statistical-analyses">Statistical analyses</h3>
<p>All statistical analyses were performed using R version 4.3.2 (2023-10-31) (http://www.r-project.org)</p>
<section id="phenotype-comparison" class="level4">
<h4 class="anchored" data-anchor-id="phenotype-comparison">Phenotype comparison</h4>
<p>Results are expressed as number of cases (%) or median [interquartile range]. Comparisons between individuals grouped according their liver steatosis were calculated using chi squared tests for categorical data and Kruskal-Wallis tests for continous variables. The significance level was set at 0.05.</p>
</section>
<section id="differential-gene-expression-analysis" class="level4">
<h4 class="anchored" data-anchor-id="differential-gene-expression-analysis">Differential gene expression analysis</h4>
</section>
<section id="whole-exome-sequencing-analysis" class="level4">
<h4 class="anchored" data-anchor-id="whole-exome-sequencing-analysis">Whole exome sequencing analysis</h4>
</section>
<section id="eqtl" class="level4">
<h4 class="anchored" data-anchor-id="eqtl">eQTL</h4>
</section>
<section id="functional-enrichment-analysis" class="level4">
<h4 class="anchored" data-anchor-id="functional-enrichment-analysis">Functional enrichment analysis</h4>
<p>Enrichment analysis for each condition, tissue and gene module was performed using the <code>gprofiler2</code> R package ( v. 0.2.3). The top pathways with an FDR-corrected p-value &lt; 0.05 were considered significantly relevant.</p>
</section>
</section>
</section>
<section id="results" class="level2">
<h2 class="anchored" data-anchor-id="results">Results</h2>
<p>This multi-omics study included 80 individuals with obesity who underwent a liver histological characterization to determine the presence of fibrosis and lipid accumulation (steatosis). Fibrosis was scored from 0 to 3 and steatosis categorized according the observed lipid content in the hepatocyte (&lt;5%, 5-33%, and &gt;33%) . Individuals´ median BMI was 45.9 kg/m<span class="math inline">\(^2\)</span>, the median age was 49 yrs (range 22–61), and 75% were women, with no observed steatosis-based differences in these parameters (as shown in <a href="#tbl-ta1" class="quarto-xref">Table&nbsp;<span>1</span></a>).</p>
<div class="cell">
<div id="tbl-ta1" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-ta1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;1: Phenotipe data. Represented as number of cases (%) or median [interquartile range]. BMI: Body mass index. <br> p.overall: p-value for the comparisons between individuals grouped according their liver steatosis (chi squared or Kruskal-Wallis tests for discrete or continous variables, respectively)
</figcaption>
<div aria-describedby="tbl-ta1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<table class="table table-striped table-condensed do-not-create-environment cell caption-top table-sm small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: center;" data-quarto-table-cell-role="th">[ALL]</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">&lt;5%</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">5-33%</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">&gt;33%</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">p.overall</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-style: italic; border-bottom: 1px solid grey;"></td>
<td style="text-align: center; font-style: italic; border-bottom: 1px solid grey;">N=80</td>
<td style="text-align: center; font-style: italic; border-bottom: 1px solid grey;">N=29</td>
<td style="text-align: center; font-style: italic; border-bottom: 1px solid grey;">N=26</td>
<td style="text-align: center; font-style: italic; border-bottom: 1px solid grey;">N=25</td>
<td style="text-align: center; font-style: italic; border-bottom: 1px solid grey;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">Reported sex:</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;">0.913</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">men</td>
<td style="text-align: center;">20 (25.0%)</td>
<td style="text-align: center;">7 (24.1%)</td>
<td style="text-align: center;">6 (23.1%)</td>
<td style="text-align: center;">7 (28.0%)</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">women</td>
<td style="text-align: center;">60 (75.0%)</td>
<td style="text-align: center;">22 (75.9%)</td>
<td style="text-align: center;">20 (76.9%)</td>
<td style="text-align: center;">18 (72.0%)</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">Age (years)</td>
<td style="text-align: center;">49.0 [41.0;55.0]</td>
<td style="text-align: center;">47.0 [41.0;52.0]</td>
<td style="text-align: center;">51.0 [42.5;56.8]</td>
<td style="text-align: center;">51.0 [41.0;54.0]</td>
<td style="text-align: center;">0.505</td>
</tr>
<tr class="even">
<td style="text-align: left;">BMI (kg/m2)</td>
<td style="text-align: center;">45.9 [41.6;50.5]</td>
<td style="text-align: center;">49.2 [41.5;51.3]</td>
<td style="text-align: center;">43.9 [41.6;50.4]</td>
<td style="text-align: center;">45.9 [42.7;46.9]</td>
<td style="text-align: center;">0.563</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Fibrosis</td>
<td style="text-align: center;">0.00 [0.00;0.50]</td>
<td style="text-align: center;">0.00 [0.00;0.00]</td>
<td style="text-align: center;">0.00 [0.00;0.75]</td>
<td style="text-align: center;">0.00 [0.00;1.00]</td>
<td style="text-align: center;">0.055</td>
</tr>
</tbody>
</table>


</div>
</div>
</figure>
</div>
</div>
</section>
<section id="discusion" class="level2">
<h2 class="anchored" data-anchor-id="discusion">Discusion</h2>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" data-line-spacing="2" role="list" style="display: none">
<div id="ref-dobin2012" class="csl-entry" role="listitem">
Dobin, A., Davis, C. A., Schlesinger, F., Drenkow, J., Zaleski, C., Jha, S., Batut, P., Chaisson, M., &amp; Gingeras, T. R. (2012). STAR: ultrafast universal RNA-seq aligner. <em>Bioinformatics</em>, <em>29</em>(1), 15–21. doi: <a href="https://doi.org/10.1093/bioinformatics/bts635">10.1093/bioinformatics/bts635</a>
</div>
<div id="ref-ewels2020" class="csl-entry" role="listitem">
Ewels, P. A., Peltzer, A., Fillinger, S., Patel, H., Alneberg, J., Wilm, A., Garcia, M. U., Di Tommaso, P., &amp; Nahnsen, S. (2020). The nf-core framework for community-curated bioinformatics pipelines. <em>Nature Biotechnology</em>, <em>38</em>(3), 276–278. doi: <a href="https://doi.org/10.1038/s41587-020-0439-x">10.1038/s41587-020-0439-x</a>
</div>
<div id="ref-garcia2020" class="csl-entry" role="listitem">
Garcia, M., Juhos, S., Larsson, M., Olason, P. I., Martin, M., Eisfeldt, J., DiLorenzo, S., Sandgren, J., Díaz De Ståhl, T., Ewels, P., Wirta, V., Nistér, M., Käller, M., &amp; Nystedt, B. (2020). Sarek: A portable workflow for whole-genome sequencing analysis of germline and somatic variants. <em>F1000Research</em>, <em>9</em>, 63. doi: <a href="https://doi.org/10.12688/f1000research.16665.2">10.12688/f1000research.16665.2</a>
</div>
<div id="ref-garrison2012" class="csl-entry" role="listitem">
Garrison, E., &amp; Marth, G. (2012). <em>Haplotype-based variant detection from short-read sequencing</em>. doi: <a href="https://doi.org/10.48550/ARXIV.1207.3907">10.48550/ARXIV.1207.3907</a>
</div>
<div id="ref-kleiner2005" class="csl-entry" role="listitem">
Kleiner, D. E., Brunt, E. M., Van Natta, M., Behling, C., Contos, M. J., Cummings, O. W., Ferrell, L. D., Liu, Y. C., Torbenson, M. S., Unalp-Arida, A., Yeh, M., McCullough, A. J., &amp; Sanyal, A. J. (2005). Design and validation of a histological scoring system for nonalcoholic fatty liver disease. <em>Hepatology</em>, <em>41</em>(6), 1313–1321. doi: <a href="https://doi.org/10.1002/hep.20701">10.1002/hep.20701</a>
</div>
<div id="ref-li2011" class="csl-entry" role="listitem">
Li, H. (2011). A statistical framework for SNP calling, mutation discovery, association mapping and population genetical parameter estimation from sequencing data. <em>Bioinformatics</em>, <em>27</em>(21), 2987–2993. doi: <a href="https://doi.org/10.1093/bioinformatics/btr509">10.1093/bioinformatics/btr509</a>
</div>
<div id="ref-martin2011" class="csl-entry" role="listitem">
Martin, M. (2011). Cutadapt removes adapter sequences from high-throughput sequencing reads. <em>EMBnet.journal</em>, <em>17</em>(1), 10. doi: <a href="https://doi.org/10.14806/ej.17.1.200">10.14806/ej.17.1.200</a>
</div>
<div id="ref-patro2017" class="csl-entry" role="listitem">
Patro, R., Duggal, G., Love, M. I., Irizarry, R. A., &amp; Kingsford, C. (2017). Salmon provides fast and bias-aware quantification of transcript expression. <em>Nature Methods</em>, <em>14</em>(4), 417–419. doi: <a href="https://doi.org/10.1038/nmeth.4197">10.1038/nmeth.4197</a>
</div>
<div id="ref-purcell2007" class="csl-entry" role="listitem">
Purcell, S., Neale, B., Todd-Brown, K., Thomas, L., Ferreira, M. A. R., Bender, D., Maller, J., Sklar, P., Bakker, P. I. W. de, Daly, M. J., &amp; Sham, P. C. (2007). PLINK: A Tool Set for Whole-Genome Association and Population-Based Linkage Analyses. <em>The American Journal of Human Genetics</em>, <em>81</em>(3), 559–575. doi: <a href="https://doi.org/10.1086/519795">10.1086/519795</a>
</div>
<div id="ref-torres-perez2015" class="csl-entry" role="listitem">
Torres-Perez, E., Valero, M., Garcia-Rodriguez, B., Gonzalez-Irazabal, Y., Calmarza, P., Calvo-Ruata, L., Ortega, C., Garcia-Sobreviela, M. P., Sanz-Paris, A., Artigas, J. M., Lagos, J., &amp; Arbones-Mainar, J. M. (2015). The FAT expandability (FATe) project: Biomarkers to determine the limit of expansion and the complications of obesity. <em>Cardiovascular Diabetology</em>, <em>14</em>(1), 1–8. doi: <a href="https://doi.org/10.1186/s12933-015-0203-6">10.1186/s12933-015-0203-6</a>
</div>
</div>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./index.html" class="pagination-link" aria-label="Front Page">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Front Page</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./wat&amp;msld.html" class="pagination-link" aria-label="Adipose tissue and MASLD">
        <span class="nav-page-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Adipose tissue and MASLD</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>