<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.4">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>2&nbsp; Bulk RNAseq – Master of Science in Omics Data Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./wes.html" rel="next">
<link href="./wat&amp;msld.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">


</head>

<body class="nav-sidebar floating slimcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./wat&amp;msld.html">Supplementary Material</a></li><li class="breadcrumb-item"><a href="./rnaseq.html"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Bulk RNAseq</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Master of Science in Omics Data Analysis</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/arbones/TFM" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Front Page</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>Manuscript</strong></span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Supplementary Material</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./wat&amp;msld.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Adipose tissue and MASLD</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./rnaseq.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Bulk RNAseq</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./wes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Whole exome sequencing</span></span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="header-section-number">2.1</span> Introduction</a></li>
  <li><a href="#experimental-design" id="toc-experimental-design" class="nav-link" data-scroll-target="#experimental-design"><span class="header-section-number">2.2</span> Experimental design</a>
  <ul class="collapse">
  <li><a href="#rna-isolation-and-extraction" id="toc-rna-isolation-and-extraction" class="nav-link" data-scroll-target="#rna-isolation-and-extraction"><span class="header-section-number">2.2.1</span> RNA isolation and extraction</a></li>
  <li><a href="#rna-quality-assessment" id="toc-rna-quality-assessment" class="nav-link" data-scroll-target="#rna-quality-assessment"><span class="header-section-number">2.2.2</span> RNA quality assessment</a></li>
  <li><a href="#stranded-mrna-library-construction-quality-control-and-sequencing" id="toc-stranded-mrna-library-construction-quality-control-and-sequencing" class="nav-link" data-scroll-target="#stranded-mrna-library-construction-quality-control-and-sequencing"><span class="header-section-number">2.2.3</span> Stranded mRNA library construction, quality control and sequencing</a></li>
  <li><a href="#mrnaseq-mapping-and-quantification-workflow" id="toc-mrnaseq-mapping-and-quantification-workflow" class="nav-link" data-scroll-target="#mrnaseq-mapping-and-quantification-workflow"><span class="header-section-number">2.2.4</span> mRNAseq mapping and quantification workflow</a></li>
  <li><a href="#quality-control-and-reads-preprocessing" id="toc-quality-control-and-reads-preprocessing" class="nav-link" data-scroll-target="#quality-control-and-reads-preprocessing"><span class="header-section-number">2.2.5</span> Quality control and reads preprocessing</a></li>
  </ul></li>
  <li><a href="#gene-expression-analysis-according-to-steatosis-using-deseq2" id="toc-gene-expression-analysis-according-to-steatosis-using-deseq2" class="nav-link" data-scroll-target="#gene-expression-analysis-according-to-steatosis-using-deseq2"><span class="header-section-number">2.3</span> Gene expression analysis according to steatosis using <code>DESeq2</code></a>
  <ul class="collapse">
  <li><a href="#data-quality-assessment" id="toc-data-quality-assessment" class="nav-link" data-scroll-target="#data-quality-assessment"><span class="header-section-number">2.3.1</span> Data quality assessment</a></li>
  <li><a href="#differential-expression-in-scwat" id="toc-differential-expression-in-scwat" class="nav-link" data-scroll-target="#differential-expression-in-scwat"><span class="header-section-number">2.3.2</span> Differential expression in scWAT</a></li>
  <li><a href="#differential-expression-in-liver" id="toc-differential-expression-in-liver" class="nav-link" data-scroll-target="#differential-expression-in-liver"><span class="header-section-number">2.3.3</span> Differential expression in liver</a></li>
  </ul></li>
  <li><a href="#bibliography" id="toc-bibliography" class="nav-link" data-scroll-target="#bibliography">Bibliography</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./wat&amp;msld.html">Supplementary Material</a></li><li class="breadcrumb-item"><a href="./rnaseq.html"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Bulk RNAseq</span></a></li></ol></nav>
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Bulk RNAseq</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="co"># libraries for visualization</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(EnhancedVolcano)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>conflicted<span class="sc">::</span><span class="fu">conflicts_prefer</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(), </span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">filter</span>(),</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>                             base<span class="sc">::</span>setdiff)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">scipen=</span><span class="dv">999</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="introduction" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="introduction"><span class="header-section-number">2.1</span> Introduction</h2>
<p>Recent advancements in omics technologies and the decreasing costs of next-generation sequencing (NGS) have transformed high-throughput data generation and analysis. Companies like Novogene and BGI, which specialize in outsourcing, have played a key role in this shift. As these technologies evolve, the large volumes of data produced daily require advanced and customizable analytical methods.</p>
<p>There are two main types of analysis tools: graphical user interface (GUI) tools and command line interface (CLI) tools. GUI tools are user-friendly and accessible to researchers without advanced programming skills, offering robust capabilities for omics data analysis. These tools can be either web-based or local applications. Web-based applications, such as Galaxy or Partek Flow, are frequently updated and provide access to the latest database versions. However, they may be less stable compared to local applications, which are more reliable for reproducible results.</p>
<p>CLI tools, on the other hand, require bioinformatics expertise but offer greater flexibility and powerful features for customizing and automating complex analyses. Programs written in Bash, R, Python, Perl, and Groovy are particularly efficient for high-throughput data processing, especially in parallel computing environments. The open-source nature of R and Python, along with their extensive libraries and packages, has made them widely adopted in bioinformatics.</p>
<p>Specialized tools for pipeline development, such as Nextflow, Snakemake, WDL, and Cromwell, have enhanced the scalability and reproducibility of bioinformatics workflows. Nextflow stands out for its ability to adapt pipelines written in various scripting languages and for its active user community that maintains and shares curated nf-core pipelines.</p>
<p>In RNA sequencing (RNAseq) data analysis, a typical pipeline involves several stages: upstream data processing, quality control, visualization of count matrices, and downstream functional analysis. Different tools excel at various stages, and integrating these tools can create a more comprehensive and customized analytical approach. This project leverages the strengths of Nextflow’s nf-core rnaseq pipeline and the versatility of R and its packages to develop a robust and flexible RNA-seq data analysis pipeline.</p>
<p>https://www.biorxiv.org/content/10.1101/2023.10.15.562329v3.full</p>
</section>
<section id="experimental-design" class="level2 page-columns page-full" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="experimental-design"><span class="header-section-number">2.2</span> Experimental design</h2>
<section id="rna-isolation-and-extraction" class="level3" data-number="2.2.1">
<h3 data-number="2.2.1" class="anchored" data-anchor-id="rna-isolation-and-extraction"><span class="header-section-number">2.2.1</span> RNA isolation and extraction</h3>
<p>Total RNA was isolated from paired frozen biopsies of scWAT (n=80) and liver (n=80) by homogenizing the tissues in 1 ml of TRIzol per sample (Sigma Aldrich, #T9424) following the manufacturer’s protocol. The homogenates were transferred to a vial and incubated for 5 minutes at room temperature to dissociate nuclear components. Next, 0.2 ml of 100% chloroform per ml of TRIzol was added, shaken vigorously, incubated for 15 minutes at room temperature, and then centrifuged at 12,000 g for 15 minutes at 4°C.</p>
<p>After centrifugation, three phases were formed. The aqueous phase containing the RNA was isolated, and 0.5 ml of 100% isopropanol was added to precipitate the RNA. The mixture was incubated on ice for 10 minutes and centrifuged again at 12,000 g for 15 minutes at 4°C. The supernatant was removed by decanting, and the RNA pellet was resuspended in 1 ml of 75% ethanol for washing. The pellet was homogenized, centrifuged at 7,500 g for 5 minutes at 4°C, and the supernatant was removed. The pellet was then dried for 5 minutes at room temperature before resuspending the RNA in DEPC water. To remove any genomic DNA, the RNA samples were treated with RNase-Free DNase (Life Technologies). The concentration and purity of the RNA were measured by absorbance at 260/280 nm and 260/230 nm using a Nanodrop 2000 (Thermo Fisher).</p>
</section>
<section id="rna-quality-assessment" class="level3 page-columns page-full" data-number="2.2.2">
<h3 data-number="2.2.2" class="anchored" data-anchor-id="rna-quality-assessment"><span class="header-section-number">2.2.2</span> RNA quality assessment</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">##### RIN numbers</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>rin <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"data/rin.csv"</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co"># histogram decoration</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>gg_hist <span class="ot">&lt;-</span> <span class="cf">function</span>(){<span class="fu">list</span>(</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">7</span>, <span class="at">fill=</span><span class="st">"darkgrey"</span>),</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"RIN"</span>,<span class="at">y=</span><span class="st">"Number of Samples</span><span class="sc">\n</span><span class="st">"</span>),</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>        <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="dv">4</span><span class="sc">:</span><span class="dv">10</span>, <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">10</span>)),</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme_minimal</span>(),</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">15</span>),</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">12</span>))</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>)} </span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>r1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(rin <span class="sc">|&gt;</span> <span class="fu">filter</span> (tissue<span class="sc">==</span><span class="st">"liver"</span>), <span class="fu">aes</span>(RIN_SCT))<span class="sc">+</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>        <span class="fu">gg_hist</span>()</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>r2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(rin <span class="sc">|&gt;</span> <span class="fu">filter</span> (tissue<span class="sc">==</span><span class="st">"scWAT"</span>), <span class="fu">aes</span>(RIN_SCT))<span class="sc">+</span></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>        <span class="fu">gg_hist</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>

<div class="no-row-height column-margin column-container"><div class="">
<div class="cell">
<div class="cell-output-display">
<div id="fig-rin" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-rin-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="rnaseq_files/figure-html/fig-rin-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-rin-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2.1: Histograms representing the distribution of RIN numbers across samples.
</figcaption>
</figure>
</div>
</div>
</div>
</div></div><p>RNA integrity was assessed using the RNA integrity number (RIN). The RIN was measured with the Agilent 2200 TapeStation and the RNA ScreenTape assay, following the manufacturer’s protocol. For liver tissue samples, the average RIN was 6.8 ± 1.0, with values ranging from 4.8 to 8.9. In contrast, scWAT samples showed higher RNA integrity, with an average RIN of 8.1 ± 0.6, ranging from 6.3 to 9.0. The higher RIN values in scWAT suggest better RNA preservation, likely due to lower enzymatic activity that degrades RNA (<a href="#fig-rin" class="quarto-xref">Figure&nbsp;<span>2.1</span></a>).</p>
</section>
<section id="stranded-mrna-library-construction-quality-control-and-sequencing" class="level3" data-number="2.2.3">
<h3 data-number="2.2.3" class="anchored" data-anchor-id="stranded-mrna-library-construction-quality-control-and-sequencing"><span class="header-section-number">2.2.3</span> Stranded mRNA library construction, quality control and sequencing</h3>
<p>The preparation of RNA libraries and sequencing was performed at the Novogene Cambridge Science Park (Cambridge, United Kingdom). Libraries were prepared using the Novogene NGS RNA Library Prep Set (PT042). Briefly, mRNA obtained with poly-T oligo-attached magnetic beads was fragmented and the first strand cDNA was synthesized using random hexamer primers. Then, the second strand cDNA was synthesized using dUTP, instead of dTTP. The directional library was ready after end repair, A-tailing, adapter ligation, enzyme digestion to remove UTP-containing second strand cDNA, size selection, PCR amplification, and purification.</p>
<p>Libraries were checked with Qubit and real-time PCR for quantification and bioanalyzer for size distribution detection. Five libraries (1 from liver and 4 from scWAT) did not meet the quality standards and were not sequenced. Libraries that passed quality controls were pooled and sequenced (2 x 150 bp) on Novaseq X Plus (Illumina), according to the effective library concentration and data amount.</p>
</section>
<section id="mrnaseq-mapping-and-quantification-workflow" class="level3" data-number="2.2.4">
<h3 data-number="2.2.4" class="anchored" data-anchor-id="mrnaseq-mapping-and-quantification-workflow"><span class="header-section-number">2.2.4</span> mRNAseq mapping and quantification workflow</h3>
<p>We used the nf-core/rnaseq pipeline with Nextflow (version 23.10.1) to process RNA sequencing data (compressed fastq files). The analysis was containerized using Docker to ensure reproducibility. Input data, including sample metadata, was provided via a CSV file, and the workflow was performed by the <a href="https://www.iacs.es/servicios/biocomputacion/">Biocomputation Core</a> on the Core Cluster of the Instituto Aragonés de Ciencias de la Salud. We employed the STAR aligner with Salmon for quantification, using the GRCh38.p13 reference genome and Gencode v40 gene annotations. The pipeline was configured to skip marking duplicate reads and using custom genome references rather than default settings. The workflow was based on the nf-core repository at <a href="https://github.com/nf-core/rnaseq"></a>.</p>
</section>
<section id="quality-control-and-reads-preprocessing" class="level3" data-number="2.2.5">
<h3 data-number="2.2.5" class="anchored" data-anchor-id="quality-control-and-reads-preprocessing"><span class="header-section-number">2.2.5</span> Quality control and reads preprocessing</h3>
<p>In our study, the initial analysis of raw sequencing data from RNAseq experiments involving liver and subcutaneous adipose tissue (scWAT) showed similar average (mean ± sd) outputs before trimming, with liver samples yielding 50.4 ± 10 million reads per sample (forward + reverse) and scWAT 52.1 ± 11 million reads. The variability in sequencing depth between samples was evidenced by ranges extending from 31.0 to 85.3 million reads per sample for liver, and 20.3 to 83.0 million reads for scWAT. (<a href="docs/multiqcReport_rnaseq-RAW.html" target="_blank">Multiqc report</a>)</p>
<p>Post-sequencing data processing with Cutadapt revealed a mean trimming percentage of 1.02 ± 0.19% for liver samples and 1.41 ± 0.44% for SCWAT, suggesting a higher incidence of adapter contamination or low-quality bases in SCWAT samples.</p>
<p>Mapping efficiencies, assessed using STAR, demonstrated robust alignment with an average of 89.6 ± 2.9% of sequences uniquely mapping to the reference genome (GRCh38) for liver samples, and a similar mapping performance 90.2 ± 1.64% for scWAT. Reads mapping to multiple loci were also similar between tissues, with 8.18 ± 1.59% for liver and 8.45 ± 1.46% for scWAT.</p>
</section>
</section>
<section id="gene-expression-analysis-according-to-steatosis-using-deseq2" class="level2 page-columns page-full" data-number="2.3">
<h2 data-number="2.3" class="anchored" data-anchor-id="gene-expression-analysis-according-to-steatosis-using-deseq2"><span class="header-section-number">2.3</span> Gene expression analysis according to steatosis using <code>DESeq2</code></h2>
<p>The analysis began by loading essential R packages: <code>tidyverse</code> <span class="citation" data-cites="tidyverse">[<a href="wat&amp;msld.html#ref-tidyverse" role="doc-biblioref">1</a>]</span>for data manipulation, <code>DESeq2</code> <span class="citation" data-cites="DESeq2">[<a href="wat&amp;msld.html#ref-DESeq2" role="doc-biblioref">2</a>]</span> for differential expression analysis, and <code>org.Hs.eg.db</code> <span class="citation" data-cites="org.Hs.eg.db">[<a href="wat&amp;msld.html#ref-org.Hs.eg.db" role="doc-biblioref">3</a>]</span> for gene annotation. Metadata were imported, and sample identifiers were established as row names. Steatosis was recategorized into three distinct groups. Gene count data were subsequently loaded, and values were rounded to the nearest integer. To enhance the robustness of the analysis, we performed a pre-filtering step to keep only those genes that had a count &gt; 15 in at least 30 samples (out of 155). Counts of the autosomal regions (PAR) were added to their homologous sequence. Gene identifiers initially provided in ENSEMBL format were converted to HGNC symbols via a mapping process that involved the creation of a dictionary through joins between the ensembl and symbol tables, linking ENSEMBL IDs to their respective HGNC counterparts. This conversion facilitated the utilization of more standardized and widely recognized gene names in subsequent analytical processes.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">###########################################</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Read metadata ########################### </span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="do">###########################################</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>meta_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"data/20240813_metadata.csv"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set the 'prefix' column as row names</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"prefix"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># recode steatosis categories into a new variable with only 3 levels</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">steatosis_3cat=</span><span class="fu">fct_recode</span>(Steatosis,<span class="st">`</span><span class="at">&gt;33%</span><span class="st">`</span><span class="ot">=</span><span class="st">"&gt;33-66%"</span>,<span class="st">`</span><span class="at">&gt;33%</span><span class="st">`</span><span class="ot">=</span><span class="st">"&gt;66%"</span>),</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">steatosis_3cat=</span><span class="fu">factor</span>(steatosis_3cat,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">"&lt;5%"</span>, <span class="st">"5-33%"</span>, <span class="st">"&gt;33%"</span>)))</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################</span></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Read count data #######################</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a><span class="do">###########################################</span></span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>counts0 <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"data/salmon.merged.gene_counts.tsv"</span>)</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>filter <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(counts0 <span class="sc">&gt;=</span> <span class="dv">30</span>) <span class="sc">&gt;=</span> <span class="dv">30</span></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>counts_filtered <span class="ot">&lt;-</span> counts0[filter,] <span class="sc">|&gt;</span> </span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">round</span>(<span class="at">digits =</span> <span class="dv">0</span>)</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a><span class="do">#########################################################</span></span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Map ENSEMBL gene IDs to HGNC symbols for compatibility with gene annotations</span></span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>symb <span class="ot">&lt;-</span> <span class="fu">mapIds</span>(org.Hs.eg.db, <span class="at">keys =</span> <span class="fu">str_sub</span>(<span class="fu">rownames</span>(counts_filtered),<span class="dv">1</span>,<span class="dv">15</span>),</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>               <span class="at">column =</span> <span class="fu">c</span>(<span class="st">"SYMBOL"</span>), <span class="at">keytype =</span> <span class="st">"ENSEMBL"</span>)</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a><span class="co"># filter out untranslated genes</span></span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>counts_na <span class="ot">&lt;-</span> counts_filtered <span class="sc">|&gt;</span> </span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">symbol =</span> symb, <span class="at">.before =</span> <span class="dv">1</span>) <span class="sc">|&gt;</span> </span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(symbol)) </span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a><span class="co"># sum the counts of duplicated genes (normaly pseudogenes)</span></span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> counts_na <span class="sc">|&gt;</span> </span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(symbol) <span class="sc">|&gt;</span> </span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), \(x) <span class="fu">sum</span>(x))) <span class="sc">|&gt;</span> </span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(symbol,<span class="at">.keep_all =</span> T) <span class="sc">|&gt;</span> </span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">|&gt;</span> </span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"symbol"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create DESeqDataSet for scWAT data</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="co"># - Select columns from the counts matrix that start with "SC_" (scWAT samples)</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># - Filter the metadata to include only scWAT samples</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co"># - Specify the design formula to include gender (sexo) and steatosis categories (steatosis_3cat)</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co"># - Indicate that the countData is not in tidy format (rows are genes, columns are samples)</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>dds_sc <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">countData =</span> counts <span class="sc">|&gt;</span> <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">"SC_"</span>)),</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">colData =</span> meta_data <span class="sc">|&gt;</span> <span class="fu">filter</span>(tissue <span class="sc">==</span> <span class="st">"scWAT"</span>),</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">design =</span> <span class="sc">~</span> sexo <span class="sc">+</span> steatosis_3cat,</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">tidy =</span> <span class="cn">FALSE</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply variance stabilizing transformation (VST) to scWAT data</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="co"># - Perform VST on the scWAT dataset without blinding to the experimental design</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>vst_sc <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds_sc, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Create DESeqDataSet for liver data</span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a><span class="co"># - Select columns from the counts matrix that start with "L_" (liver samples)</span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a><span class="co"># - Filter the metadata to include only liver samples</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a><span class="co"># - Specify the design formula to include gender (sexo) and steatosis categories (steatosis_3cat)</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>dds_liver <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(</span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">countData =</span> counts <span class="sc">|&gt;</span> <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">"L_"</span>)),</span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">colData =</span> meta_data <span class="sc">|&gt;</span> <span class="fu">filter</span>(tissue <span class="sc">==</span> <span class="st">"liver"</span>),</span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">design =</span> <span class="sc">~</span> sexo <span class="sc">+</span> steatosis_3cat</span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply variance stabilizing transformation (VST) to liver data</span></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a><span class="co"># - Perform VST on the liver dataset without blinding to the experimental design</span></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>vst_liver <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds_liver, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="data-quality-assessment" class="level3 page-columns page-full" data-number="2.3.1">
<h3 data-number="2.3.1" class="anchored" data-anchor-id="data-quality-assessment"><span class="header-section-number">2.3.1</span> Data quality assessment</h3>
<p>Data quality assessment and quality control are pivotal initial steps in any RNAseq analysis, particularly aimed at ensuring the fitness of data for detecting differentially expressed genes. These procedures are designed to identify and exclude samples and or data points compromised by experimental anomalies, thereby safeguarding the integrity of the analysis.</p>
<p>Boxplots in <a href="#fig-raw_counts1" class="quarto-xref">Figure&nbsp;<span>2.2</span></a> display log2-transformed raw counts grouped by tissue and steatosis severity levels. The data across steatosis categories (“&lt;5%”, “5-33%”, “&gt;33%”) are visualized with varying colors (gree, orange, and red) to highlight differences in expression profiles. In both scWAT and liver all samples showed similar dispersion and central tendencies in gene expression counts. The uniformity in distribution across all steatosis levels suggested that library depth and average counts per sample were consistent, precluding any library bias in the analysis.</p>
<div class="cell page-columns page-full">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="do">########## scWAT ##############################################################</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="do">###############################################################################</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract counts data from DESeq2 object for scWAT and convert to a data frame</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Access raw counts data from the DESeq2 object (dds_sc)</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>counts_sc <span class="ot">&lt;-</span> dds_sc<span class="sc">@</span>assays<span class="sc">@</span>data<span class="sc">$</span>counts <span class="sc">|&gt;</span> </span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span> </span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Move gene names into a column named "gene"</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Reshape the data from wide to long format with gene, sample, and counts columns</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="st">"gene"</span>, <span class="at">names_to =</span> <span class="st">"sample"</span>, <span class="at">values_to =</span> <span class="st">"counts"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Join the counts data with metadata to add sample information</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(meta_data <span class="sc">|&gt;</span> <span class="fu">rownames_to_column</span>(<span class="st">"prefix"</span>), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"sample"</span> <span class="ot">=</span> <span class="st">"prefix"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Ensure sample names are treated as characters</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample.y =</span> <span class="fu">as.character</span>(sample.y))</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Function for boxplot decoration</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>gg_box <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add a boxplot geometry, remove missing values (na.rm = TRUE)</span></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Apply a log2 transformation to the y-axis</span></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">"log2"</span>),</span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add labels for the x and y axes</span></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"</span><span class="sc">\n</span><span class="st">Sample"</span>, <span class="at">y =</span> <span class="st">"Raw counts"</span>),</span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set custom fill colors for steatosis categories</span></span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="st">"Steatosis"</span>, <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#99c140"</span>, <span class="st">"#e7b416"</span>, <span class="st">"#db7b2b"</span>)),</span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Flip the coordinates to make the boxplots horizontal</span></span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>(),</span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Apply a minimal theme for a clean appearance</span></span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>(),</span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Adjust the size of the axis titles</span></span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>))</span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Boxplot visualizing the log-transformed read counts across samples grouped by steatosis categories</span></span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(counts_sc, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(sample.y, <span class="fu">as.numeric</span>(steatosis_3cat)),</span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a>                            <span class="co"># Add 1 to counts for log transformation, and fill by steatosis category</span></span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a>                            <span class="at">y =</span> counts <span class="sc">+</span> <span class="dv">1</span>, <span class="at">fill =</span> steatosis_3cat)) <span class="sc">+</span></span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Apply the gg_box function for consistent styling</span></span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true" tabindex="-1"></a>      <span class="fu">gg_box</span>()</span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-45"><a href="#cb5-45" aria-hidden="true" tabindex="-1"></a><span class="do">########## liver ##############################################################</span></span>
<span id="cb5-46"><a href="#cb5-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-47"><a href="#cb5-47" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract counts data from DESeq2 object for liver and convert to a data frame</span></span>
<span id="cb5-48"><a href="#cb5-48" aria-hidden="true" tabindex="-1"></a><span class="co"># Access raw counts data from the DESeq2 object (dds_liver)</span></span>
<span id="cb5-49"><a href="#cb5-49" aria-hidden="true" tabindex="-1"></a>counts_liver <span class="ot">&lt;-</span> dds_liver<span class="sc">@</span>assays<span class="sc">@</span>data<span class="sc">$</span>counts <span class="sc">|&gt;</span> </span>
<span id="cb5-50"><a href="#cb5-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span> </span>
<span id="cb5-51"><a href="#cb5-51" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Move gene names into a column named "gene"</span></span>
<span id="cb5-52"><a href="#cb5-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb5-53"><a href="#cb5-53" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Reshape the data from wide to long format with gene, sample, and counts columns</span></span>
<span id="cb5-54"><a href="#cb5-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="st">"gene"</span>, <span class="at">names_to =</span> <span class="st">"sample"</span>, <span class="at">values_to =</span> <span class="st">"counts"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb5-55"><a href="#cb5-55" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Join the counts data with metadata to add sample information</span></span>
<span id="cb5-56"><a href="#cb5-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(meta_data <span class="sc">|&gt;</span> <span class="fu">rownames_to_column</span>(<span class="st">"prefix"</span>), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"sample"</span> <span class="ot">=</span> <span class="st">"prefix"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb5-57"><a href="#cb5-57" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Ensure sample names are treated as characters</span></span>
<span id="cb5-58"><a href="#cb5-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample.y =</span> <span class="fu">as.character</span>(sample.y))</span>
<span id="cb5-59"><a href="#cb5-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-60"><a href="#cb5-60" aria-hidden="true" tabindex="-1"></a><span class="co"># Boxplot visualizing the log-transformed read counts across samples grouped by steatosis categories</span></span>
<span id="cb5-61"><a href="#cb5-61" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(counts_liver, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(sample.y, <span class="fu">as.numeric</span>(steatosis_3cat)),</span>
<span id="cb5-62"><a href="#cb5-62" aria-hidden="true" tabindex="-1"></a>                               <span class="co"># Add 1 to counts for log transformation, and fill by steatosis category</span></span>
<span id="cb5-63"><a href="#cb5-63" aria-hidden="true" tabindex="-1"></a>                               <span class="at">y =</span> counts <span class="sc">+</span> <span class="dv">1</span>, <span class="at">fill =</span> steatosis_3cat)) <span class="sc">+</span></span>
<span id="cb5-64"><a href="#cb5-64" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Apply the gg_box function for consistence</span></span>
<span id="cb5-65"><a href="#cb5-65" aria-hidden="true" tabindex="-1"></a>      <span class="fu">gg_box</span>()</span>
<span id="cb5-66"><a href="#cb5-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-67"><a href="#cb5-67" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine the two boxplots (scWAT and liver) into a single figure with labeled panels</span></span>
<span id="cb5-68"><a href="#cb5-68" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(p1, p2, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">'a'</span>, <span class="st">'b'</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display page-columns page-full">
<div id="fig-raw_counts1" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-raw_counts1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="page-columns page-full">
<img src="rnaseq_files/figure-html/fig-raw_counts1-1.png" class="img-fluid figure-img column-body-outset-right" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-raw_counts1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2.2: Boxplots representing the distribution of log2-transformed read counts across samples (a: scWAT, b: liver), with distinct coloration representing different steatosis categories. Each box represents the interquartile range (IQR) of the counts, with the line inside the box denoting the median. The whiskers extend to show the range of the data, excluding outliers, which are plotted as individual points
</figcaption>
</figure>
</div>
</div>
</div>
<p>To further observe the distribution of samples, read counts across samples were normalized using the variance stabilizing transformation (VST) <span class="citation" data-cites="tibshirani1988">[<a href="wat&amp;msld.html#ref-tibshirani1988" role="doc-biblioref">4</a>]</span> and their dimension was reduced via principal component analysis (PCA) and visualized the results within combinations of the two principal components. The biplot (<a href="#fig-pca1" class="quarto-xref">Figure&nbsp;<span>2.3</span></a>) shows tight clusters for both scWAT (a) and liver (b) samples. This clustering indicates that samples within the same group (based on steatosis level or gender) have similar expression profiles, suggesting strong internal consistency.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate PCA for scWAT data using "sexo" and "steatosis_3cat" as factors, return PCA data</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>pca_data_sc <span class="ot">&lt;-</span> <span class="fu">plotPCA</span>(vst_sc, <span class="fu">c</span>(<span class="st">"sexo"</span>, <span class="st">"steatosis_3cat"</span>), <span class="at">returnData =</span> <span class="cn">TRUE</span>)   </span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate and round the percentage of variance explained by the first two principal components (PC1 and PC2)</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>percentVar <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> <span class="fu">attr</span>(pca_data_sc, <span class="st">"percentVar"</span>))    </span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a scatter plot for the first two principal components (PC1 and PC2) for scWAT data</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_data_sc, <span class="fu">aes</span>(PC1, PC2, <span class="at">color =</span> steatosis_3cat, <span class="at">shape =</span> sexo)) <span class="sc">+</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>   </span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">"PC1: "</span>, percentVar[<span class="dv">1</span>], <span class="st">"% variance"</span>)) <span class="sc">+</span>  </span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">"PC2: "</span>, percentVar[<span class="dv">2</span>], <span class="st">"% variance"</span>)) <span class="sc">+</span>   </span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>() <span class="sc">+</span>    <span class="co"># Set aspect ratio to be equal</span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="st">"Steatosis"</span>, <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#99c140"</span>,<span class="st">"#e7b416"</span>,<span class="st">"#db7b2b"</span>)) <span class="sc">+</span>   </span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span>  </span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)    <span class="co"># Hide legend</span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate PCA for liver data using "sexo" and "steatosis_3cat" as factors, return PCA data</span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>pca_data_liver <span class="ot">&lt;-</span> <span class="fu">plotPCA</span>(vst_liver, <span class="fu">c</span>(<span class="st">"sexo"</span>, <span class="st">"steatosis_3cat"</span>), <span class="at">returnData =</span> <span class="cn">TRUE</span>)   </span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate and round the percentage of variance explained by PC1 and PC2 for liver data</span></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>percentVar <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> <span class="fu">attr</span>(pca_data_liver, <span class="st">"percentVar"</span>))    </span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a scatter plot for the first two principal components (PC1 and PC2) for liver data</span></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_data_liver, <span class="fu">aes</span>(PC1, PC2, <span class="at">color =</span> steatosis_3cat, <span class="at">shape =</span> sexo)) <span class="sc">+</span></span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>    </span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">"PC1: "</span>, percentVar[<span class="dv">1</span>], <span class="st">"% variance"</span>)) <span class="sc">+</span>    </span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">"PC2: "</span>, percentVar[<span class="dv">2</span>], <span class="st">"% variance"</span>)) <span class="sc">+</span>   </span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="st">"Steatosis"</span>, <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#99c140"</span>,<span class="st">"#e7b416"</span>,<span class="st">"#db7b2b"</span>)) <span class="sc">+</span>    </span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>() <span class="sc">+</span>    <span class="co"># Set aspect ratio to be equal</span></span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()    </span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine the two PCA plots (scWAT and liver) into a single figure with labeled panels</span></span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(p3, p4, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">'scWAT'</span>, <span class="st">'Liver'</span>))    </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-pca1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pca1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="rnaseq_files/figure-html/fig-pca1-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pca1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2.3: PCA representation of normalized counts
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="differential-expression-in-scwat" class="level3 page-columns page-full" data-number="2.3.2">
<h3 data-number="2.3.2" class="anchored" data-anchor-id="differential-expression-in-scwat"><span class="header-section-number">2.3.2</span> Differential expression in scWAT</h3>
<section id="modeling-raw-counts" class="level4" data-number="2.3.2.1">
<h4 data-number="2.3.2.1" class="anchored" data-anchor-id="modeling-raw-counts"><span class="header-section-number">2.3.2.1</span> Modeling raw counts</h4>
<p>In our study, the <code>DESeq2</code> package was employed to analyze differential gene expression across scWAT samples with varying degrees of liver steatosis. Initially, we configured the <code>DESeq2</code> object using the Likelihood Ratio Test (LRT) method, suitable for handling complex comparisons involving multiple levels of a factor.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform differential expression analysis for scWAT data</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Run DESeq2 using the likelihood ratio test (LRT) controlling for the effect of gender (sexo)</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>dex_sc <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds_sc, <span class="at">test =</span> <span class="st">"LRT"</span>, <span class="at">reduced =</span> <span class="sc">~</span> sexo)    </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Under the hood of the <code>DESeq</code> function the normalization of count data was performed using the median of ratios method, ensuring that variations in library size did not bias the analysis. This method adjusted each sample’s read counts by a size factor, calculated to equalize the median of ratios across all samples (<a href="#fig-fs_sc" class="quarto-xref">Figure&nbsp;<span>2.4</span></a>).</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Extract and visualize size factors from the DESeq2 object</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sizeFactors</span>(dex_sc) <span class="sc">|&gt;</span> <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span> </span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"Sample"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">`</span><span class="at">Size Factors</span><span class="st">`</span><span class="ot">=</span> <span class="st">"sizeFactors(dex_sc)"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample =</span> <span class="fu">str_sub</span> (Sample, <span class="dv">4</span>,<span class="dv">6</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>Sample, <span class="at">y=</span> <span class="st">`</span><span class="at">Size Factors</span><span class="st">`</span>))<span class="sc">+</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">col=</span><span class="st">"darkgrey"</span>, <span class="at">fill =</span> <span class="st">"grey"</span>)<span class="sc">+</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()<span class="sc">+</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-fs_sc" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-fs_sc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="rnaseq_files/figure-html/fig-fs_sc-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-fs_sc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2.4: Calculated factor sizes for each sample using the median of ratios method.
</figcaption>
</figure>
</div>
</div>
</div>
<p>To address the intrinsic variability in count data, <code>DESeq2</code> estimates gene-wise dispersion, which reflects the variance deviation from the mean for each gene. This step is crucial because RNA-seq data typically exhibit variance that exceeds the mean, particularly when expression levels are low. Dispersion parameters were calculated for each gene (Black dots in <a href="#fig-disp_sc" class="quarto-xref">Figure&nbsp;<span>2.5</span></a>), taking into account the mean expression levels and observed variance.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate a plot of dispersion estimates using the DESeq2 object</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>d2 <span class="ot">&lt;-</span> <span class="fu">plotDispEsts</span>(dex_sc)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-disp_sc" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-disp_sc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="rnaseq_files/figure-html/fig-disp_sc-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-disp_sc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2.5: Dispersion plot. Black dots represents single genes and the intial maximum likelihood dispersion estimate (MLE) given the observed data, the red line is the fitted curve to gene-wise dispersion estimates, while blue dots are the shrunken gene-wise dispersion estimates.
</figcaption>
</figure>
</div>
</div>
</div>
<p>The gene-wise dispersions were then modeled through a curve-fitting process, to generalize dispersion behavior across the dataset. The curve, represented in red in <a href="#fig-disp_sc" class="quarto-xref">Figure&nbsp;<span>2.5</span></a>, illustrates the inverse relationship between mean expression and dispersion, a typical observation in RNA-seq data. The final analytical step involved shrinking the initial raw dispersion estimates towards the values predicted by the fitted curve (Blue dots).</p>
<p>Our analysis revealed a pattern where dispersion generally decreased with increasing mean expression levels, indicating a stable and consistent data quality across our sample set. The limited shrinkage observed was attributable to the large number of biological replicates, which provided a robust basis for the dispersion estimates.</p>
</section>
<section id="testing-for-differential-expression" class="level4 page-columns page-full" data-number="2.3.2.2">
<h4 data-number="2.3.2.2" class="anchored" data-anchor-id="testing-for-differential-expression"><span class="header-section-number">2.3.2.2</span> Testing for differential expression</h4>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use DESeq2 to extract results from the differential expression analysis with FDR adjustment</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>res_sc <span class="ot">&lt;-</span> <span class="fu">results</span>(dex_sc, <span class="at">pAdjustMethod =</span> <span class="st">"fdr"</span>, <span class="at">independentFiltering =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert the results to a data frame for easier manipulation</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>res_sc_df <span class="ot">&lt;-</span> res_sc <span class="sc">|&gt;</span> </span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter the results to identify significantly differentially expressed genes (DEGs)</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep only those genes with an adjusted p-value (padj) less than 0.05</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>degs_sc <span class="ot">&lt;-</span> res_sc_df <span class="sc">|&gt;</span> </span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Count the number of upregulated genes (log2FoldChange &gt; 0)</span></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>up <span class="ot">&lt;-</span> <span class="fu">sum</span>(res_sc_df<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Count the number of downregulated genes (log2FoldChange &lt; 0)</span></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>down <span class="ot">&lt;-</span> <span class="fu">sum</span>(res_sc_df<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>In our differential expression analysis using <code>DESeq2</code> with the LRT model, we initially identified 9849 genes as upregulated and 4682 as downregulated in the subcutaneous white adipose tissue (scWAT) of individuals with the highest steatosis compared to those with no steatosis. However, after applying a stringent false discovery rate (FDR) cutoff of less than 0.05, the number of statistically significant differentially expressed genes (DEGs) was greatly reduced to 20. Of these, only 2 genes were found to be upregulated, while 18 were downregulated (<a href="#tbl-degs_sc" class="quarto-xref">Table&nbsp;<span>2.1</span></a>)</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select relevant columns from the significant DEGs in scWAT data</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>degs_sc <span class="sc">|&gt;</span> </span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Select the log2 fold change, p-value, and adjusted p-value (FDR)</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(log2FoldChange, pvalue, <span class="at">fdr =</span> <span class="st">"padj"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Arrange the results by the FDR (adjusted p-value) in ascending order</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(fdr) <span class="sc">|&gt;</span> </span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create a formatted table with specified number of digits for each column</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">5</span>, <span class="dv">3</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Apply striped styling to the table for better visual presentation</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="st">"striped"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="tbl-degs_sc" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-degs_sc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;2.1: Differentially expressed genes (DEGs) in scWAT. fdr: false discovery rate
</figcaption>
<div aria-describedby="tbl-degs_sc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<table class="table table-striped do-not-create-environment cell caption-top table-sm small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th">log2FoldChange</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">pvalue</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">fdr</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">HOOK3</td>
<td style="text-align: right;">-1.41</td>
<td style="text-align: right;">0.00000</td>
<td style="text-align: right;">0.000</td>
</tr>
<tr class="even">
<td style="text-align: left;">ZNF429</td>
<td style="text-align: right;">-1.07</td>
<td style="text-align: right;">0.00000</td>
<td style="text-align: right;">0.002</td>
</tr>
<tr class="odd">
<td style="text-align: left;">USHBP1</td>
<td style="text-align: right;">-0.78</td>
<td style="text-align: right;">0.00000</td>
<td style="text-align: right;">0.010</td>
</tr>
<tr class="even">
<td style="text-align: left;">POLRMT</td>
<td style="text-align: right;">-0.40</td>
<td style="text-align: right;">0.00002</td>
<td style="text-align: right;">0.027</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CETP</td>
<td style="text-align: right;">-1.09</td>
<td style="text-align: right;">0.00002</td>
<td style="text-align: right;">0.027</td>
</tr>
<tr class="even">
<td style="text-align: left;">ZNF500</td>
<td style="text-align: right;">-0.63</td>
<td style="text-align: right;">0.00002</td>
<td style="text-align: right;">0.027</td>
</tr>
<tr class="odd">
<td style="text-align: left;">PEX6</td>
<td style="text-align: right;">-0.57</td>
<td style="text-align: right;">0.00003</td>
<td style="text-align: right;">0.027</td>
</tr>
<tr class="even">
<td style="text-align: left;">PDE11A</td>
<td style="text-align: right;">-1.04</td>
<td style="text-align: right;">0.00003</td>
<td style="text-align: right;">0.027</td>
</tr>
<tr class="odd">
<td style="text-align: left;">OSER1-DT</td>
<td style="text-align: right;">-1.00</td>
<td style="text-align: right;">0.00002</td>
<td style="text-align: right;">0.027</td>
</tr>
<tr class="even">
<td style="text-align: left;">ABHD18</td>
<td style="text-align: right;">-0.66</td>
<td style="text-align: right;">0.00003</td>
<td style="text-align: right;">0.027</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SEMA3C</td>
<td style="text-align: right;">0.61</td>
<td style="text-align: right;">0.00002</td>
<td style="text-align: right;">0.027</td>
</tr>
<tr class="even">
<td style="text-align: left;">LYRM7</td>
<td style="text-align: right;">-0.94</td>
<td style="text-align: right;">0.00001</td>
<td style="text-align: right;">0.027</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DCLRE1C</td>
<td style="text-align: right;">-0.58</td>
<td style="text-align: right;">0.00003</td>
<td style="text-align: right;">0.027</td>
</tr>
<tr class="even">
<td style="text-align: left;">WDR73</td>
<td style="text-align: right;">-0.75</td>
<td style="text-align: right;">0.00004</td>
<td style="text-align: right;">0.030</td>
</tr>
<tr class="odd">
<td style="text-align: left;">INTS7</td>
<td style="text-align: right;">0.72</td>
<td style="text-align: right;">0.00005</td>
<td style="text-align: right;">0.035</td>
</tr>
<tr class="even">
<td style="text-align: left;">NAT8L</td>
<td style="text-align: right;">-0.60</td>
<td style="text-align: right;">0.00008</td>
<td style="text-align: right;">0.047</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SAC3D1</td>
<td style="text-align: right;">-0.67</td>
<td style="text-align: right;">0.00007</td>
<td style="text-align: right;">0.047</td>
</tr>
<tr class="even">
<td style="text-align: left;">GATD1-DT</td>
<td style="text-align: right;">-0.83</td>
<td style="text-align: right;">0.00008</td>
<td style="text-align: right;">0.047</td>
</tr>
<tr class="odd">
<td style="text-align: left;">INTS1</td>
<td style="text-align: right;">-0.62</td>
<td style="text-align: right;">0.00008</td>
<td style="text-align: right;">0.047</td>
</tr>
<tr class="even">
<td style="text-align: left;">CCDC61</td>
<td style="text-align: right;">-0.62</td>
<td style="text-align: right;">0.00009</td>
<td style="text-align: right;">0.050</td>
</tr>
</tbody>
</table>


</div>
</div>
</figure>
</div>
</div>
<p>In this regard, the volcano plot (<a href="#fig-volcano_sc" class="quarto-xref">Figure&nbsp;<span>2.6</span></a>) illustrates the differential expression of genes in subcutaneous white adipose tissue (scWAT) according to liver steatosis. Among the thousands of genes analyzed, those genes displayed in red showed both statistically significant p-values (FDR &lt; 0.05) and substantial changes in expression (Log2 FC &gt; 1).</p>
<div class="cell page-columns page-full">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a volcano plot for differentially expressed genes (DEGs) in scWAT</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_sc,</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Label the points on the plot with gene names from the rownames of res_sc</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="fu">rownames</span>(res_sc),</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set the main title of the plot</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Differentially expressed genes (DEGs) in scWAT"</span>,</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set the subtitle indicating the comparison groups</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"&gt;33%  vs. &lt;5% steatosis"</span>,</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Customize the legend labels for different categories of DEGs</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabels =</span> <span class="fu">c</span>(<span class="st">"NS"</span>, <span class="fu">expression</span>(Log[<span class="dv">2</span>] <span class="sc">~</span> FC  <span class="sc">&gt;</span> <span class="dv">1</span>), <span class="st">"FDR &lt; 0.05"</span>, <span class="fu">expression</span>( FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">~</span> and <span class="sc">~</span> Log[<span class="dv">2</span>] <span class="sc">~</span> FC  <span class="sc">&gt;</span> <span class="dv">1</span>)),</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set the caption below the plot (empty in this case)</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">""</span>,</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set the size of the labels on the plot</span></span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="dv">4</span>,</span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Position the legend to the right of the plot</span></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'right'</span>,</span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Draw lines connecting labels to their corresponding points</span></span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Connect labels to points in both directions (from left and right)</span></span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">directionConnectors =</span> <span class="st">"both"</span>,</span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Allow for an infinite number of overlapping points to be labeled</span></span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">max.overlaps =</span> <span class="cn">Inf</span>,</span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set the fold change cutoff for highlighting DEGs</span></span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">1</span>,</span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Specify the column to use for the p-value cutoff</span></span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoffCol =</span> <span class="st">"padj"</span>,</span>
<span id="cb12-27"><a href="#cb12-27" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set the p-value cutoff for significance</span></span>
<span id="cb12-28"><a href="#cb12-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb12-29"><a href="#cb12-29" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set the x-axis to represent log2 fold change</span></span>
<span id="cb12-30"><a href="#cb12-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,</span>
<span id="cb12-31"><a href="#cb12-31" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set the y-axis to represent the p-value</span></span>
<span id="cb12-32"><a href="#cb12-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'pvalue'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display page-columns page-full">
<div id="fig-volcano_sc" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-volcano_sc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="page-columns page-full">
<img src="rnaseq_files/figure-html/fig-volcano_sc-1.png" class="img-fluid figure-img column-body-outset-right" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-volcano_sc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2.6: Volcano plot showing DEGs (in red) in individuals with the highest (&gt;33%) compared to the lowest (&lt;5%) degree of steatosis.
</figcaption>
</figure>
</div>
</div>
</div>
<p>Following their identification, we further analyzed the expression patterns of these 20 DEGs by representing their normalized counts in boxplots across the three steatosis groups (<a href="#fig-box_sc" class="quarto-xref">Figure&nbsp;<span>2.7</span></a>). This approach allows us to observe how the expression of these significant genes varies across different degrees of steatosis, offering deeper insights into the potential role these genes play in the progression of steatosis. The boxplots highlight the consistency of expression changes across groups, reinforcing the relevance of these DEGs in the context of steatosis severity.</p>
<div class="cell page-columns page-full">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalize the counts for scWAT data and transpose the matrix</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>counts_norm_sc <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">counts</span>(dex_sc, <span class="at">normalized =</span> <span class="cn">TRUE</span>))</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset the metadata for scWAT to match the normalized counts</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>meta_data_sc <span class="ot">&lt;-</span> meta_data[<span class="fu">rownames</span>(counts_norm_sc), <span class="fu">c</span>(<span class="st">"sexo"</span>, <span class="st">"steatosis_3cat"</span>)]</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a data frame combining normalized counts with metadata</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert the transposed counts to a data frame</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>sc <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(counts_norm_sc) <span class="sc">|&gt;</span> </span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Select columns corresponding to the DEGs identified in scWAT</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">rownames</span>(degs_sc)) <span class="sc">|&gt;</span> </span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Bind the selected DEGs with the corresponding metadata</span></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_cols</span>(meta_data_sc) <span class="sc">|&gt;</span> </span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Reshape the data to long format for ggplot, with Gene and Normalized Counts columns</span></span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(degs_sc), <span class="at">names_to =</span> <span class="st">"Gene"</span>, <span class="at">values_to =</span> <span class="st">"Normalized Counts"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add 1 to avoid infinity in the log scale </span></span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Normalized Counts</span><span class="st">`</span><span class="ot">=</span><span class="st">`</span><span class="at">Normalized Counts</span><span class="st">`</span> <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Function for decorating boxplots of DEGs</span></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>gg_degs <span class="ot">&lt;-</span> <span class="cf">function</span>() { </span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(</span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Apply a log2 transformation to the y-axis (normalized counts)</span></span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">"log2"</span>),</span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set the x-axis label to "Steatosis"</span></span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"</span><span class="sc">\n</span><span class="st">Steatosis"</span>),</span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add a boxplot geometry with specified width and no outliers</span></span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.5</span>, <span class="at">outliers =</span> <span class="cn">FALSE</span>),</span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add jittered points for individual observations with transparency</span></span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.25</span>, <span class="at">size =</span> <span class="dv">2</span>, <span class="at">alpha =</span> <span class="fl">0.3</span>),</span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add statistical comparison (Kruskal-Wallis test) labels to the plot</span></span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_compare_means</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">"p= "</span>, ..p.format..)), </span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="st">"kruskal.test"</span>,</span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a>                       <span class="at">label.x.npc =</span> <span class="fl">0.3</span>,</span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a>                       <span class="at">label.y.npc =</span> <span class="fl">0.8</span>,</span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a>                       <span class="at">size =</span> <span class="dv">4</span>),</span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set custom fill colors for the steatosis categories</span></span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#99c140"</span>, <span class="st">"#e7b416"</span>, <span class="st">"#db7b2b"</span>)),</span>
<span id="cb13-40"><a href="#cb13-40" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Apply a minimal theme for clean appearance</span></span>
<span id="cb13-41"><a href="#cb13-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>(),</span>
<span id="cb13-42"><a href="#cb13-42" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create facets for each gene, allowing free scaling on the y-axis, with 4 columns of facets</span></span>
<span id="cb13-43"><a href="#cb13-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span> Gene, <span class="at">scales =</span> <span class="st">"free_y"</span>, <span class="at">ncol =</span> <span class="dv">4</span>),</span>
<span id="cb13-44"><a href="#cb13-44" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Customize theme elements: remove legend, adjust text sizes, and rotate x-axis labels</span></span>
<span id="cb13-45"><a href="#cb13-45" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb13-46"><a href="#cb13-46" aria-hidden="true" tabindex="-1"></a>          <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">13</span>),</span>
<span id="cb13-47"><a href="#cb13-47" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">13</span>),</span>
<span id="cb13-48"><a href="#cb13-48" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">11</span>),</span>
<span id="cb13-49"><a href="#cb13-49" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">angle =</span> <span class="dv">20</span>))</span>
<span id="cb13-50"><a href="#cb13-50" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb13-51"><a href="#cb13-51" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-52"><a href="#cb13-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-53"><a href="#cb13-53" aria-hidden="true" tabindex="-1"></a><span class="do">##############################################</span></span>
<span id="cb13-54"><a href="#cb13-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-55"><a href="#cb13-55" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the normalized counts across steatosis categories for each DEG in scWAT</span></span>
<span id="cb13-56"><a href="#cb13-56" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(sc, <span class="fu">aes</span>(<span class="at">x =</span> steatosis_3cat, <span class="at">y =</span> <span class="st">`</span><span class="at">Normalized Counts</span><span class="st">`</span>, <span class="at">fill =</span> steatosis_3cat)) <span class="sc">+</span></span>
<span id="cb13-57"><a href="#cb13-57" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Apply the custom ggplot decorations for DEGs</span></span>
<span id="cb13-58"><a href="#cb13-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gg_degs</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display page-columns page-full">
<div id="fig-box_sc" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-box_sc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="page-columns page-full">
<img src="rnaseq_files/figure-html/fig-box_sc-1.png" class="img-fluid figure-img column-body-outset-right" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-box_sc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2.7: Boxplots of the differentially expressed genes (DEGs) in scWAT according to the 3 groups of steatosis.<br>Each box represents the interquartile range (IQR) of the normalized counts, with the line inside the box denoting the median. The whiskers extend to show the range of the data, excluding outliers, which are plotted as individual points.<br>p: p-value for the Kruskal-Wallis test for the comparison between groups.
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="gene-ontology-enrichment" class="level4 page-columns page-full" data-number="2.3.2.3">
<h4 data-number="2.3.2.3" class="anchored" data-anchor-id="gene-ontology-enrichment"><span class="header-section-number">2.3.2.3</span> Gene ontology enrichment</h4>
<p>Lastly, in our RNAseq analysis of subcutaneous adipose tissue (scWAT) from individuals with varying degrees of steatosis, we identified specific molecular functions associated with selected DEGs through gene ontology (GO) analysis. The <a href="#tbl-go_sc" class="quarto-xref">Table&nbsp;<span>2.2</span></a> summarizes those molecular functions significantly enriched . Thus, semaphorin receptor binding and chemorepellent activity were up-regulated in the steatotic conditions, while multiple functions related to fatty acid metabolism, lipid transport, microtubule binding, cyclic nucleotide signaling, and nucleic acid processing where appeared down-regulated. This enrichment analysis sheds light on the molecular functions most impacted by steatosis and provides a deeper understanding of the biological processes that are differentially regulated in scWAT as steatosis progresses.</p>
<div class="cell page-columns page-full">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Gene Ontology (Molecular Function) </span><span class="al">###</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Flag up- and down-regulated genes based on adjusted p-value and log2 fold change</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>go_sc <span class="ot">&lt;-</span> res_sc <span class="sc">|&gt;</span> </span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">q_0.1 =</span> <span class="fu">case_when</span>(</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>    padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">~</span> <span class="dv">1</span>,    <span class="co"># Flag genes as upregulated (1) if padj &lt; 0.05 and log2FoldChange &gt; 0</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>    padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">~</span> <span class="sc">-</span><span class="dv">1</span>,   <span class="co"># Flag genes as downregulated (-1) if padj &lt; 0.05 and log2FoldChange &lt; 0</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">.default =</span> <span class="dv">0</span>                             <span class="co"># All other genes are flagged as 0 (not significantly regulated)</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract upregulated genes</span></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>go_sc_up_00  <span class="ot">&lt;-</span> go_sc <span class="sc">|&gt;</span> </span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(q_0<span class="fl">.1</span> <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">|&gt;</span>           <span class="co"># Filter for upregulated genes</span></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(gene)                      <span class="co"># Pull the gene names into a vector</span></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform GO enrichment analysis for upregulated genes</span></span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>go_sc_up_0 <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> go_sc_up_00, <span class="at">OrgDb =</span> <span class="st">"org.Hs.eg.db"</span>, <span class="at">keyType =</span> <span class="st">"SYMBOL"</span>, <span class="at">ont =</span> <span class="st">"MF"</span>, </span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">qvalueCutoff =</span> <span class="fl">0.05</span>, <span class="at">pAdjustMethod =</span> <span class="st">"fdr"</span>)    <span class="co"># Enrichment analysis for molecular function (MF)</span></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert the result to a data frame and add a regulation label ("Up")</span></span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>go_sc_up <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(go_sc_up_0) <span class="sc">|&gt;</span> </span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">regulation =</span> <span class="st">"Up"</span>)</span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract downregulated genes</span></span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>go_sc_down_00  <span class="ot">&lt;-</span> go_sc <span class="sc">|&gt;</span> </span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(q_0<span class="fl">.1</span> <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span>) <span class="sc">|&gt;</span>          <span class="co"># Filter for downregulated genes</span></span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(gene)                      <span class="co"># Pull the gene names into a vector</span></span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform GO enrichment analysis for downregulated genes</span></span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a>go_sc_down_0 <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> go_sc_down_00, <span class="at">OrgDb =</span> <span class="st">"org.Hs.eg.db"</span>, <span class="at">keyType =</span> <span class="st">"SYMBOL"</span>, <span class="at">ont =</span> <span class="st">"MF"</span>, </span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>  <span class="at">pvalueCutoff =</span> <span class="fl">0.1</span>, <span class="at">qvalueCutoff =</span> <span class="fl">0.05</span>, <span class="at">pAdjustMethod =</span> <span class="st">"fdr"</span>)    <span class="co"># Enrichment analysis for molecular function (MF)</span></span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert the result to a data frame and add a regulation label ("Down")</span></span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>go_sc_down <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(go_sc_down_0) <span class="sc">|&gt;</span> </span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">regulation =</span> <span class="st">"Down"</span>)</span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a table combining up- and down-regulated genes and their GO terms</span></span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a><span class="fu">bind_rows</span>(go_sc_up, go_sc_down) <span class="sc">|&gt;</span> </span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(regulation), geneID) <span class="sc">|&gt;</span>  <span class="co"># Arrange by regulation status and geneID</span></span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(regulation, geneID, <span class="st">`</span><span class="at">GO molecular function</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Description"</span>, <span class="st">`</span><span class="at">GO ID</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"ID"</span>, qvalue ) <span class="sc">|&gt;</span>  <span class="co"># Select relevant columns</span></span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">digits=</span><span class="dv">3</span>) <span class="sc">|&gt;</span>                     <span class="co"># Format the table with 3 decimal places</span></span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="st">"striped"</span>)             <span class="co"># Apply striped styling to the table</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="tbl-go_sc" class="cell quarto-float quarto-figure quarto-figure-center anchored page-columns page-full">
<figure class="quarto-float quarto-float-tbl figure page-columns page-full">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-go_sc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;2.2: Gene ontology (GO) analysis to detect enriched molecular functions of up- and down-regulated genes in individuals with the highest (&gt;33%) compared to the lowest (&lt;5%) degree of steatosis.
</figcaption>
<div aria-describedby="tbl-go_sc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="page-columns page-full">
<div class="cell-output-display column-body-outset-right">
<table class="table table-striped do-not-create-environment cell caption-top table-sm small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">regulation</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">geneID</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">GO molecular function</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">GO ID</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">qvalue</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">GO:0030215</td>
<td style="text-align: left;">Up</td>
<td style="text-align: left;">SEMA3C</td>
<td style="text-align: left;">semaphorin receptor binding</td>
<td style="text-align: left;">GO:0030215</td>
<td style="text-align: right;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0045499</td>
<td style="text-align: left;">Up</td>
<td style="text-align: left;">SEMA3C</td>
<td style="text-align: left;">chemorepellent activity</td>
<td style="text-align: left;">GO:0045499</td>
<td style="text-align: right;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0008017</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CCDC61/HOOK3</td>
<td style="text-align: left;">microtubule binding</td>
<td style="text-align: left;">GO:0008017</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0015631</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CCDC61/HOOK3</td>
<td style="text-align: left;">tubulin binding</td>
<td style="text-align: left;">GO:0015631</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0120020</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CETP</td>
<td style="text-align: left;">cholesterol transfer activity</td>
<td style="text-align: left;">GO:0120020</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0120015</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CETP</td>
<td style="text-align: left;">sterol transfer activity</td>
<td style="text-align: left;">GO:0120015</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0031210</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CETP</td>
<td style="text-align: left;">phosphatidylcholine binding</td>
<td style="text-align: left;">GO:0031210</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0015248</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CETP</td>
<td style="text-align: left;">sterol transporter activity</td>
<td style="text-align: left;">GO:0015248</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0015485</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CETP</td>
<td style="text-align: left;">cholesterol binding</td>
<td style="text-align: left;">GO:0015485</td>
<td style="text-align: right;">0.025</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0050997</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CETP</td>
<td style="text-align: left;">quaternary ammonium group binding</td>
<td style="text-align: left;">GO:0050997</td>
<td style="text-align: right;">0.025</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0120013</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CETP</td>
<td style="text-align: left;">lipid transfer activity</td>
<td style="text-align: left;">GO:0120013</td>
<td style="text-align: right;">0.025</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0032934</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CETP</td>
<td style="text-align: left;">sterol binding</td>
<td style="text-align: left;">GO:0032934</td>
<td style="text-align: right;">0.027</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0005548</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CETP</td>
<td style="text-align: left;">phospholipid transporter activity</td>
<td style="text-align: left;">GO:0005548</td>
<td style="text-align: right;">0.027</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0043178</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CETP</td>
<td style="text-align: left;">alcohol binding</td>
<td style="text-align: left;">GO:0043178</td>
<td style="text-align: right;">0.032</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0005496</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CETP</td>
<td style="text-align: left;">steroid binding</td>
<td style="text-align: left;">GO:0005496</td>
<td style="text-align: right;">0.034</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0000014</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">DCLRE1C</td>
<td style="text-align: left;">single-stranded DNA endodeoxyribonuclease activity</td>
<td style="text-align: left;">GO:0000014</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0008409</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">DCLRE1C</td>
<td style="text-align: left;">5'-3' exonuclease activity</td>
<td style="text-align: left;">GO:0008409</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0016895</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">DCLRE1C</td>
<td style="text-align: left;">DNA exonuclease activity, producing 5'-phosphomonoesters</td>
<td style="text-align: left;">GO:0016895</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0004529</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">DCLRE1C</td>
<td style="text-align: left;">DNA exonuclease activity</td>
<td style="text-align: left;">GO:0004529</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0004520</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">DCLRE1C</td>
<td style="text-align: left;">DNA endonuclease activity</td>
<td style="text-align: left;">GO:0004520</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0004536</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">DCLRE1C</td>
<td style="text-align: left;">DNA nuclease activity</td>
<td style="text-align: left;">GO:0004536</td>
<td style="text-align: right;">0.027</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0003684</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">DCLRE1C</td>
<td style="text-align: left;">damaged DNA binding</td>
<td style="text-align: left;">GO:0003684</td>
<td style="text-align: right;">0.028</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0034452</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">HOOK3</td>
<td style="text-align: left;">dynactin binding</td>
<td style="text-align: left;">GO:0034452</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0051959</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">HOOK3</td>
<td style="text-align: left;">dynein light intermediate chain binding</td>
<td style="text-align: left;">GO:0051959</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0045505</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">HOOK3</td>
<td style="text-align: left;">dynein intermediate chain binding</td>
<td style="text-align: left;">GO:0045505</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0044183</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">LYRM7</td>
<td style="text-align: left;">protein folding chaperone</td>
<td style="text-align: left;">GO:0044183</td>
<td style="text-align: right;">0.027</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0008080</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">NAT8L</td>
<td style="text-align: left;">N-acetyltransferase activity</td>
<td style="text-align: left;">GO:0008080</td>
<td style="text-align: right;">0.027</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0016410</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">NAT8L</td>
<td style="text-align: left;">N-acyltransferase activity</td>
<td style="text-align: left;">GO:0016410</td>
<td style="text-align: right;">0.033</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0016407</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">NAT8L</td>
<td style="text-align: left;">acetyltransferase activity</td>
<td style="text-align: left;">GO:0016407</td>
<td style="text-align: right;">0.033</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0030553</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">PDE11A</td>
<td style="text-align: left;">cGMP binding</td>
<td style="text-align: left;">GO:0030553</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0047555</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">PDE11A</td>
<td style="text-align: left;">3',5'-cyclic-GMP phosphodiesterase activity</td>
<td style="text-align: left;">GO:0047555</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0004115</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">PDE11A</td>
<td style="text-align: left;">3',5'-cyclic-AMP phosphodiesterase activity</td>
<td style="text-align: left;">GO:0004115</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0004114</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">PDE11A</td>
<td style="text-align: left;">3',5'-cyclic-nucleotide phosphodiesterase activity</td>
<td style="text-align: left;">GO:0004114</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0004112</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">PDE11A</td>
<td style="text-align: left;">cyclic-nucleotide phosphodiesterase activity</td>
<td style="text-align: left;">GO:0004112</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0030551</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">PDE11A</td>
<td style="text-align: left;">cyclic nucleotide binding</td>
<td style="text-align: left;">GO:0030551</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0008081</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">PDE11A</td>
<td style="text-align: left;">phosphoric diester hydrolase activity</td>
<td style="text-align: left;">GO:0008081</td>
<td style="text-align: right;">0.032</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0140036</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">PEX6</td>
<td style="text-align: left;">ubiquitin-dependent protein binding</td>
<td style="text-align: left;">GO:0140036</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0140035</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">PEX6</td>
<td style="text-align: left;">ubiquitination-like modification-dependent protein binding</td>
<td style="text-align: left;">GO:0140035</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0140318</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">PEX6</td>
<td style="text-align: left;">protein transporter activity</td>
<td style="text-align: left;">GO:0140318</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0003899</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">POLRMT</td>
<td style="text-align: left;">DNA-directed 5'-3' RNA polymerase activity</td>
<td style="text-align: left;">GO:0003899</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0034062</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">POLRMT</td>
<td style="text-align: left;">5'-3' RNA polymerase activity</td>
<td style="text-align: left;">GO:0034062</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0097747</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">POLRMT</td>
<td style="text-align: left;">RNA polymerase activity</td>
<td style="text-align: left;">GO:0097747</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0000175</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">POLRMT</td>
<td style="text-align: left;">3'-5'-RNA exonuclease activity</td>
<td style="text-align: left;">GO:0000175</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0016896</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">POLRMT</td>
<td style="text-align: left;">RNA exonuclease activity, producing 5'-phosphomonoesters</td>
<td style="text-align: left;">GO:0016896</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0004532</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">POLRMT</td>
<td style="text-align: left;">RNA exonuclease activity</td>
<td style="text-align: left;">GO:0004532</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0008408</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">POLRMT</td>
<td style="text-align: left;">3'-5' exonuclease activity</td>
<td style="text-align: left;">GO:0008408</td>
<td style="text-align: right;">0.025</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0016796</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">POLRMT/DCLRE1C</td>
<td style="text-align: left;">exonuclease activity, active with either ribo- or deoxyribonucleic acids and producing 5'-phosphomonoesters</td>
<td style="text-align: left;">GO:0016796</td>
<td style="text-align: right;">0.018</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0004527</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">POLRMT/DCLRE1C</td>
<td style="text-align: left;">exonuclease activity</td>
<td style="text-align: left;">GO:0004527</td>
<td style="text-align: right;">0.018</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0004518</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">POLRMT/DCLRE1C</td>
<td style="text-align: left;">nuclease activity</td>
<td style="text-align: left;">GO:0004518</td>
<td style="text-align: right;">0.022</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0030165</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">USHBP1</td>
<td style="text-align: left;">PDZ domain binding</td>
<td style="text-align: left;">GO:0030165</td>
<td style="text-align: right;">0.032</td>
</tr>
</tbody>
</table>


</div>
</div>
</figure>
</div>
</div>
</section>
</section>
<section id="differential-expression-in-liver" class="level3 page-columns page-full" data-number="2.3.3">
<h3 data-number="2.3.3" class="anchored" data-anchor-id="differential-expression-in-liver"><span class="header-section-number">2.3.3</span> Differential expression in liver</h3>
<section id="modeling-raw-counts-1" class="level4" data-number="2.3.3.1">
<h4 data-number="2.3.3.1" class="anchored" data-anchor-id="modeling-raw-counts-1"><span class="header-section-number">2.3.3.1</span> Modeling raw counts</h4>
<p>For this study we also investigated the differential gene expression across liver biopsies with varying degrees of steatosis. To this end the <code>DESeq2</code> object was also set using the Likelihood Ratio Test (LRT) method.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform differential expression analysis for liver data</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Run DESeq2 using the likelihood ratio test (LRT) controlling for the effect of gender (sexo)</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>dex_liver <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds_liver,<span class="at">test =</span> <span class="st">"LRT"</span>,  <span class="at">reduced =</span> <span class="sc">~</span> sexo)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>In the liver tissue analysis, the first step involved calculating the size factor of the libraries to equalize the median of ratios across all samples (<a href="#fig-fs_liver" class="quarto-xref">Figure&nbsp;<span>2.8</span></a>).</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract and visualize size factors from the DESeq2 object</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sizeFactors</span>(dex_liver) <span class="sc">|&gt;</span> <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span> </span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"Sample"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">`</span><span class="at">Size Factors</span><span class="st">`</span><span class="ot">=</span> <span class="st">"sizeFactors(dex_liver)"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample =</span> <span class="fu">str_sub</span> (Sample, <span class="dv">4</span>,<span class="dv">6</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>Sample, <span class="at">y=</span> <span class="st">`</span><span class="at">Size Factors</span><span class="st">`</span>))<span class="sc">+</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">col=</span><span class="st">"darkgrey"</span>, <span class="at">fill =</span> <span class="st">"grey"</span>)<span class="sc">+</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()<span class="sc">+</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-fs_liver" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-fs_liver-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="rnaseq_files/figure-html/fig-fs_liver-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-fs_liver-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2.8: Calculated factor sizes for each sample using the median of ratios method.
</figcaption>
</figure>
</div>
</div>
</div>
<p>We next assessed the variation in gene expression in liver tissue by modeling gene-wise dispersions through a curve-fitting process. Consistent with our observations in the scWAT analysis, the liver data revealed that dispersion generally decreased as mean expression levels increased (<a href="#fig-disp_liver" class="quarto-xref">Figure&nbsp;<span>2.9</span></a>). The limited shrinkage of dispersion estimates, as seen in scWAT, can be attributed to the large number of biological replicates.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate a plot of dispersion estimates using the DESeq2 object</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>d2 <span class="ot">&lt;-</span> <span class="fu">plotDispEsts</span>(dex_liver)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-disp_liver" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-disp_liver-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="rnaseq_files/figure-html/fig-disp_liver-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-disp_liver-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2.9: Dispersion plot. Black dots represents single genes and the intial maximum likelihood dispersion estimate (MLE) given the observed data, the red line is the fitted curve to gene-wise dispersion estimates, while blue dots are the shrunken gene-wise dispersion estimates.
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="testing-for-differential-expression-1" class="level4 page-columns page-full" data-number="2.3.3.2">
<h4 data-number="2.3.3.2" class="anchored" data-anchor-id="testing-for-differential-expression-1"><span class="header-section-number">2.3.3.2</span> Testing for differential expression</h4>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use DESeq2 to extract results from the differential expression analysis with FDR adjustment</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>res_liver <span class="ot">&lt;-</span> <span class="fu">results</span>(dex_liver, <span class="at">pAdjustMethod =</span> <span class="st">"fdr"</span>, <span class="at">independentFiltering =</span> <span class="cn">TRUE</span>)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert the results to a data frame for easier manipulation</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>res_liver_df <span class="ot">&lt;-</span> res_liver <span class="sc">|&gt;</span> </span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter the results to identify significantly differentially expressed genes (DEGs)</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep only those genes with an adjusted p-value (padj) less than 0.05</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>degs_liver <span class="ot">&lt;-</span> res_liver_df <span class="sc">|&gt;</span> </span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Count the number of upregulated genes (log2FoldChange &gt; 0)</span></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>up <span class="ot">&lt;-</span> <span class="fu">sum</span>(res_liver_df<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Count the number of downregulated genes (log2FoldChange &lt; 0)</span></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>down <span class="ot">&lt;-</span> <span class="fu">sum</span>(res_liver_df<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>In our differential expression analysis using <code>DESeq2</code> with the LRT model, we initially identified 7748 genes as upregulated and 6810 as downregulated in the liver of individuals with the highest steatosis compared to those with no steatosis. However, when applying a stringent false discovery rate (FDR) cutoff of less than 0.05 to identify statistically significant differentially expressed genes (DEGs), the number of significant DEGs was markedly reduced to 12. Among these, only 1 genes were found to be upregulated, while 11 were downregulated (<a href="#tbl-degs_liver" class="quarto-xref">Table&nbsp;<span>2.3</span></a>). The volcano plot (<a href="#fig-volcano_liver" class="quarto-xref">Figure&nbsp;<span>2.10</span></a>) illustrates the differential expression of genes in the liver according to steatosis. Among the thousands of genes analyzed, only IGFBP2 showed FDR &lt; 0.05 and substantial changes in expression (Log2 FC &gt; 1).</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select relevant columns from the significant DEGs in liver data</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>degs_liver <span class="sc">|&gt;</span> </span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Select the log2 fold change, p-value, and adjusted p-value (FDR)</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(log2FoldChange, pvalue, <span class="at">fdr =</span> <span class="st">"padj"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Arrange the results by the FDR (adjusted p-value) in ascending order</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(fdr) <span class="sc">|&gt;</span> </span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create a formatted table with specified number of digits for each column</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">5</span>, <span class="dv">3</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Apply striped styling to the table for better visual presentation</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="st">"striped"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="tbl-degs_liver" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-degs_liver-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;2.3: Differentially expressed genes (DEGs) in liver. fdr: false discovery rate
</figcaption>
<div aria-describedby="tbl-degs_liver-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<table class="table table-striped do-not-create-environment cell caption-top table-sm small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th">log2FoldChange</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">pvalue</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">fdr</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">IL32</td>
<td style="text-align: right;">0.81</td>
<td style="text-align: right;">0.00000</td>
<td style="text-align: right;">0.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">IGFBP2</td>
<td style="text-align: right;">-1.37</td>
<td style="text-align: right;">0.00001</td>
<td style="text-align: right;">0.008</td>
</tr>
<tr class="odd">
<td style="text-align: left;">AHCY</td>
<td style="text-align: right;">-0.59</td>
<td style="text-align: right;">0.00003</td>
<td style="text-align: right;">0.014</td>
</tr>
<tr class="even">
<td style="text-align: left;">RN7SL1</td>
<td style="text-align: right;">-0.72</td>
<td style="text-align: right;">0.00003</td>
<td style="text-align: right;">0.014</td>
</tr>
<tr class="odd">
<td style="text-align: left;">RN7SL2</td>
<td style="text-align: right;">-0.69</td>
<td style="text-align: right;">0.00012</td>
<td style="text-align: right;">0.024</td>
</tr>
<tr class="even">
<td style="text-align: left;">APOF</td>
<td style="text-align: right;">-0.56</td>
<td style="text-align: right;">0.00014</td>
<td style="text-align: right;">0.024</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CYP3A4</td>
<td style="text-align: right;">-0.63</td>
<td style="text-align: right;">0.00014</td>
<td style="text-align: right;">0.024</td>
</tr>
<tr class="even">
<td style="text-align: left;">SERPINA1</td>
<td style="text-align: right;">-0.55</td>
<td style="text-align: right;">0.00009</td>
<td style="text-align: right;">0.024</td>
</tr>
<tr class="odd">
<td style="text-align: left;">AKR7A3</td>
<td style="text-align: right;">-0.67</td>
<td style="text-align: right;">0.00014</td>
<td style="text-align: right;">0.024</td>
</tr>
<tr class="even">
<td style="text-align: left;">MAT1A</td>
<td style="text-align: right;">-0.35</td>
<td style="text-align: right;">0.00011</td>
<td style="text-align: right;">0.024</td>
</tr>
<tr class="odd">
<td style="text-align: left;">PC</td>
<td style="text-align: right;">-0.43</td>
<td style="text-align: right;">0.00011</td>
<td style="text-align: right;">0.024</td>
</tr>
<tr class="even">
<td style="text-align: left;">SERPINA3</td>
<td style="text-align: right;">-0.48</td>
<td style="text-align: right;">0.00023</td>
<td style="text-align: right;">0.036</td>
</tr>
</tbody>
</table>


</div>
</div>
</figure>
</div>
</div>
<div class="cell page-columns page-full">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a volcano plot for differentially expressed genes (DEGs) in liver</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_liver,</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Label the points on the plot with gene names from the rownames of res_liver</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="fu">rownames</span>(res_liver),</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set the main title of the plot</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Differentially expressed genes (DEGs) in liver"</span>,</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set the subtitle indicating the comparison groups</span></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"&gt;33%  vs. &lt;5% steatosis"</span>,</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Customize the legend labels for different categories of DEGs</span></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabels =</span> <span class="fu">c</span>(<span class="st">"NS"</span>, <span class="fu">expression</span>(Log[<span class="dv">2</span>] <span class="sc">~</span> FC  <span class="sc">&gt;</span> <span class="dv">1</span>), <span class="st">"FDR &lt; 0.05"</span>, <span class="fu">expression</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">~</span> and <span class="sc">~</span> Log[<span class="dv">2</span>] <span class="sc">~</span> FC  <span class="sc">&gt;</span> <span class="dv">1</span>)),</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set the caption below the plot (empty in this case)</span></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">""</span>,</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Highlight only the selected DEGs in the plot</span></span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">selectLab =</span> <span class="fu">rownames</span>(degs_liver),</span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set the size of the labels on the plot</span></span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="dv">4</span>,</span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Position the legend to the right of the plot</span></span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'right'</span>,</span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Draw lines connecting labels to their corresponding points</span></span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Connect labels to points in both directions (from left and right)</span></span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">directionConnectors =</span> <span class="st">"both"</span>,</span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Allow for an infinite number of overlapping points to be labeled</span></span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">max.overlaps =</span> <span class="cn">Inf</span>,</span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set the fold change cutoff for highlighting DEGs</span></span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">1</span>,</span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Specify the column to use for the p-value cutoff</span></span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoffCol =</span> <span class="st">"padj"</span>,</span>
<span id="cb20-29"><a href="#cb20-29" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set the p-value cutoff for significance</span></span>
<span id="cb20-30"><a href="#cb20-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb20-31"><a href="#cb20-31" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set the x-axis to represent log2 fold change</span></span>
<span id="cb20-32"><a href="#cb20-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,</span>
<span id="cb20-33"><a href="#cb20-33" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set the y-axis to represent the p-value</span></span>
<span id="cb20-34"><a href="#cb20-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'pvalue'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display page-columns page-full">
<div id="fig-volcano_liver" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-volcano_liver-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="page-columns page-full">
<img src="rnaseq_files/figure-html/fig-volcano_liver-1.png" class="img-fluid figure-img column-body-outset-right" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-volcano_liver-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2.10: Volcano plot showing liver DEGs (in red) in individuals with the highest (&gt;33%) compared to the lowest (&lt;5%) degree of steatosis.
</figcaption>
</figure>
</div>
</div>
</div>
<p>After identifying the DEGs in the liver, we analyzed their expression patterns using boxplots across the three steatosis groups (<a href="#fig-box_liver" class="quarto-xref">Figure&nbsp;<span>2.11</span></a>). Most of these boxplots revealed a clear, ordered, and stepwise pattern of gene expression, with expression levels consistently and progressively changing as steatosis severity increased from low (&lt;5%) to medium, and then to high (&gt;33%). This graded response suggests that these genes are closely linked to the progression of steatosis, highlighting their potential role as key markers or drivers in the development of the condition.</p>
<div class="cell page-columns page-full">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalize the counts for liver data and transpose the matrix</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>counts_norm_liver <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">counts</span>(dex_liver, <span class="at">normalized =</span> <span class="cn">TRUE</span>))</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset the metadata for liver to match the normalized counts</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>meta_data_liver <span class="ot">&lt;-</span> meta_data[<span class="fu">rownames</span>(counts_norm_liver), <span class="fu">c</span>(<span class="st">"sexo"</span>, <span class="st">"steatosis_3cat"</span>)]</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a data frame combining normalized counts with metadata</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert the transposed counts to a data frame</span></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>liver <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(counts_norm_liver) <span class="sc">|&gt;</span> </span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Select columns corresponding to the DEGs identified in liver data</span></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">rownames</span>(degs_liver)) <span class="sc">|&gt;</span> </span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Bind the selected DEGs with the corresponding metadata</span></span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_cols</span>(meta_data_liver) <span class="sc">|&gt;</span> </span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Reshape the data to long format for ggplot, with Gene and Normalized Counts columns</span></span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(degs_liver), <span class="at">names_to =</span> <span class="st">"Gene"</span>, <span class="at">values_to =</span> <span class="st">"Normalized Counts"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add 1 to the normalized counts to avoid issues with log transformation</span></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Normalized Counts</span><span class="st">`</span> <span class="ot">=</span> <span class="st">`</span><span class="at">Normalized Counts</span><span class="st">`</span> <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the normalized counts across steatosis categories for each DEG in liver</span></span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(liver, <span class="fu">aes</span>(<span class="at">x =</span> steatosis_3cat, <span class="at">y =</span> <span class="st">`</span><span class="at">Normalized Counts</span><span class="st">`</span>, <span class="at">fill =</span> steatosis_3cat)) <span class="sc">+</span></span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Apply the custom ggplot decorations for DEGs (using the gg_degs function defined earlier)</span></span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gg_degs</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display page-columns page-full">
<div id="fig-box_liver" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-box_liver-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="page-columns page-full">
<img src="rnaseq_files/figure-html/fig-box_liver-1.png" class="img-fluid figure-img column-body-outset-right" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-box_liver-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2.11: Boxplots of the differentially expressed genes (DEGs) in liver according to the 3 groups of steatosis.<br>Each box represents the interquartile range (IQR) of the normalized counts, with the line inside the box denoting the median. The whiskers extend to show the range of the data, excluding outliers, which are plotted as individual points.<br>p: p-value for the Kruskal-Wallis test for the comparison between groups.
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="gene-ontology-enrichment-1" class="level4 page-columns page-full" data-number="2.3.3.3">
<h4 data-number="2.3.3.3" class="anchored" data-anchor-id="gene-ontology-enrichment-1"><span class="header-section-number">2.3.3.3</span> Gene ontology enrichment</h4>
<p>Lastly, in our RNAseq analysis of liver tissue from individuals with varying degrees of steatosis, we identified specific molecular functions associated with selected DEGs through gene ontology (GO) enrichment analysis. The <a href="#tbl-go_liver" class="quarto-xref">Table&nbsp;<span>2.4</span></a> summarizes the molecular functions that were significantly enriched. For instance, cytokine activity was upregulated, as indicated by the enrichment of <em>IL32</em>. In contrast, various functions related to oxidative stress, lipid metabolism, and enzyme activity, such as aldo-keto reductase activity, cholesterol binding, steroid binding, and oxidoreductase activity, were downregulated, involving genes like <em>AKR7A3</em>, <em>APOF</em>, and <em>CYP3A4</em>. Additionally, functions related to protease binding and growth factor binding, linked to genes such as <em>SERPINA1</em>, <em>SERPINA3</em>, and <em>IGFBP2</em>, were also downregulated. This GO enrichment analysis provides insights into the molecular functions most affected by steatosis in liver tissue, offering a deeper understanding of the biological processes that are differentially regulated as steatosis progresses.</p>
<div class="cell page-columns page-full">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Gene Ontology (Molecular Function) </span><span class="al">###</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Flag up- and down-regulated genes based on adjusted p-value and log2 fold change</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>go_liver <span class="ot">&lt;-</span> res_liver <span class="sc">|&gt;</span> </span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">q_0.1 =</span> <span class="fu">case_when</span>(</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>    padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">~</span> <span class="dv">1</span>,    <span class="co"># Flag genes as upregulated (1) if padj &lt; 0.05 and log2FoldChange &gt; 0</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>    padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">~</span> <span class="sc">-</span><span class="dv">1</span>,   <span class="co"># Flag genes as downregulated (-1) if padj &lt; 0.05 and log2FoldChange &lt; 0</span></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">.default =</span> <span class="dv">0</span>                             <span class="co"># All other genes are flagged as 0 (not significantly regulated)</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract upregulated genes</span></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>go_liver_up_00  <span class="ot">&lt;-</span> go_liver <span class="sc">|&gt;</span> </span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(q_0<span class="fl">.1</span> <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">|&gt;</span>           <span class="co"># Filter for upregulated genes</span></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(gene)                      <span class="co"># Pull the gene names into a vector</span></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform GO enrichment analysis for upregulated genes</span></span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>go_liver_up_0 <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> go_liver_up_00, <span class="at">OrgDb =</span> <span class="st">"org.Hs.eg.db"</span>, <span class="at">keyType =</span> <span class="st">"SYMBOL"</span>, <span class="at">ont =</span> <span class="st">"MF"</span>, </span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">qvalueCutoff =</span> <span class="fl">0.05</span>, <span class="at">pAdjustMethod =</span> <span class="st">"fdr"</span>)    <span class="co"># Enrichment analysis for molecular function (MF)</span></span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert the result to a data frame and add a regulation label ("Up")</span></span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a>go_liver_up <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(go_liver_up_0) <span class="sc">|&gt;</span> </span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">regulation =</span> <span class="st">"Up"</span>)</span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract downregulated genes</span></span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a>go_liver_down_00  <span class="ot">&lt;-</span> go_liver <span class="sc">|&gt;</span> </span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(q_0<span class="fl">.1</span> <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span>) <span class="sc">|&gt;</span>          <span class="co"># Filter for downregulated genes</span></span>
<span id="cb22-30"><a href="#cb22-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(gene)                      <span class="co"># Pull the gene names into a vector</span></span>
<span id="cb22-31"><a href="#cb22-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-32"><a href="#cb22-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform GO enrichment analysis for downregulated genes</span></span>
<span id="cb22-33"><a href="#cb22-33" aria-hidden="true" tabindex="-1"></a>go_liver_down_0 <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> go_liver_down_00, <span class="at">OrgDb =</span> <span class="st">"org.Hs.eg.db"</span>, <span class="at">keyType =</span> <span class="st">"SYMBOL"</span>, <span class="at">ont =</span> <span class="st">"MF"</span>, </span>
<span id="cb22-34"><a href="#cb22-34" aria-hidden="true" tabindex="-1"></a>  <span class="at">pvalueCutoff =</span> <span class="fl">0.1</span>, <span class="at">qvalueCutoff =</span> <span class="fl">0.05</span>, <span class="at">pAdjustMethod =</span> <span class="st">"fdr"</span>)    <span class="co"># Enrichment analysis for molecular function (MF)</span></span>
<span id="cb22-35"><a href="#cb22-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-36"><a href="#cb22-36" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert the result to a data frame and add a regulation label ("Down")</span></span>
<span id="cb22-37"><a href="#cb22-37" aria-hidden="true" tabindex="-1"></a>go_liver_down <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(go_liver_down_0) <span class="sc">|&gt;</span> </span>
<span id="cb22-38"><a href="#cb22-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">regulation =</span> <span class="st">"Down"</span>)</span>
<span id="cb22-39"><a href="#cb22-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-40"><a href="#cb22-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a table combining up- and down-regulated genes and their GO terms</span></span>
<span id="cb22-41"><a href="#cb22-41" aria-hidden="true" tabindex="-1"></a><span class="fu">bind_rows</span>(go_liver_up, go_liver_down) <span class="sc">|&gt;</span> </span>
<span id="cb22-42"><a href="#cb22-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(regulation), geneID) <span class="sc">|&gt;</span>  <span class="co"># Arrange by regulation status and geneID</span></span>
<span id="cb22-43"><a href="#cb22-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(regulation, geneID, <span class="st">`</span><span class="at">GO molecular function</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Description"</span>, <span class="st">`</span><span class="at">GO ID</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"ID"</span>, qvalue ) <span class="sc">|&gt;</span>  <span class="co"># Select relevant columns</span></span>
<span id="cb22-44"><a href="#cb22-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">digits=</span><span class="dv">3</span>) <span class="sc">|&gt;</span>                     <span class="co"># Format the table with 3 decimal places</span></span>
<span id="cb22-45"><a href="#cb22-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="st">"striped"</span>)             <span class="co"># Apply striped styling to the table</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="tbl-go_liver" class="cell quarto-float quarto-figure quarto-figure-center anchored page-columns page-full">
<figure class="quarto-float quarto-float-tbl figure page-columns page-full">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-go_liver-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;2.4: Gene ontology (GO) analysis to detect enriched molecular functions of up- and down-regulated genes in individuals with the highest (&gt;33%) compared to the lowest (&lt;5%) degree of steatosis.
</figcaption>
<div aria-describedby="tbl-go_liver-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="page-columns page-full">
<div class="cell-output-display column-body-outset-right">
<table class="table table-striped do-not-create-environment cell caption-top table-sm small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">regulation</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">geneID</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">GO molecular function</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">GO ID</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">qvalue</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">GO:0005125</td>
<td style="text-align: left;">Up</td>
<td style="text-align: left;">IL32</td>
<td style="text-align: left;">cytokine activity</td>
<td style="text-align: left;">GO:0005125</td>
<td style="text-align: right;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0004033</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">AKR7A3</td>
<td style="text-align: left;">aldo-keto reductase (NADP) activity</td>
<td style="text-align: left;">GO:0004033</td>
<td style="text-align: right;">0.017</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0009055</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">AKR7A3</td>
<td style="text-align: left;">electron transfer activity</td>
<td style="text-align: left;">GO:0009055</td>
<td style="text-align: right;">0.038</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0016616</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">AKR7A3</td>
<td style="text-align: left;">oxidoreductase activity, acting on the CH-OH group of donors, NAD or NADP as acceptor</td>
<td style="text-align: left;">GO:0016616</td>
<td style="text-align: right;">0.038</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0016614</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">AKR7A3</td>
<td style="text-align: left;">oxidoreductase activity, acting on CH-OH group of donors</td>
<td style="text-align: left;">GO:0016614</td>
<td style="text-align: right;">0.038</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0015485</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">APOF</td>
<td style="text-align: left;">cholesterol binding</td>
<td style="text-align: left;">GO:0015485</td>
<td style="text-align: right;">0.023</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0032934</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">APOF</td>
<td style="text-align: left;">sterol binding</td>
<td style="text-align: left;">GO:0032934</td>
<td style="text-align: right;">0.025</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0043178</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">APOF</td>
<td style="text-align: left;">alcohol binding</td>
<td style="text-align: left;">GO:0043178</td>
<td style="text-align: right;">0.033</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0005319</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">APOF</td>
<td style="text-align: left;">lipid transporter activity</td>
<td style="text-align: left;">GO:0005319</td>
<td style="text-align: right;">0.040</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0005496</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">APOF/CYP3A4</td>
<td style="text-align: left;">steroid binding</td>
<td style="text-align: left;">GO:0005496</td>
<td style="text-align: right;">0.009</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0008401</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CYP3A4</td>
<td style="text-align: left;">retinoic acid 4-hydroxylase activity</td>
<td style="text-align: left;">GO:0008401</td>
<td style="text-align: right;">0.011</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0016725</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CYP3A4</td>
<td style="text-align: left;">oxidoreductase activity, acting on CH or CH2 groups</td>
<td style="text-align: left;">GO:0016725</td>
<td style="text-align: right;">0.011</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0008395</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CYP3A4</td>
<td style="text-align: left;">steroid hydroxylase activity</td>
<td style="text-align: left;">GO:0008395</td>
<td style="text-align: right;">0.017</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0070330</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CYP3A4</td>
<td style="text-align: left;">aromatase activity</td>
<td style="text-align: left;">GO:0070330</td>
<td style="text-align: right;">0.017</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0016709</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CYP3A4</td>
<td style="text-align: left;">oxidoreductase activity, acting on paired donors, with incorporation or reduction of molecular oxygen, NAD(P)H as one donor, and incorporation of one atom of oxygen</td>
<td style="text-align: left;">GO:0016709</td>
<td style="text-align: right;">0.020</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0019825</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CYP3A4</td>
<td style="text-align: left;">oxygen binding</td>
<td style="text-align: left;">GO:0019825</td>
<td style="text-align: right;">0.020</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0016712</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CYP3A4</td>
<td style="text-align: left;">oxidoreductase activity, acting on paired donors, with incorporation or reduction of molecular oxygen, reduced flavin or flavoprotein as one donor, and incorporation of one atom of oxygen</td>
<td style="text-align: left;">GO:0016712</td>
<td style="text-align: right;">0.021</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0004497</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CYP3A4</td>
<td style="text-align: left;">monooxygenase activity</td>
<td style="text-align: left;">GO:0004497</td>
<td style="text-align: right;">0.036</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0020037</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CYP3A4</td>
<td style="text-align: left;">heme binding</td>
<td style="text-align: left;">GO:0020037</td>
<td style="text-align: right;">0.038</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0046906</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CYP3A4</td>
<td style="text-align: left;">tetrapyrrole binding</td>
<td style="text-align: left;">GO:0046906</td>
<td style="text-align: right;">0.038</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0005506</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CYP3A4</td>
<td style="text-align: left;">iron ion binding</td>
<td style="text-align: left;">GO:0005506</td>
<td style="text-align: right;">0.038</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0016705</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">CYP3A4</td>
<td style="text-align: left;">oxidoreductase activity, acting on paired donors, with incorporation or reduction of molecular oxygen</td>
<td style="text-align: left;">GO:0016705</td>
<td style="text-align: right;">0.040</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0031994</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">IGFBP2</td>
<td style="text-align: left;">insulin-like growth factor I binding</td>
<td style="text-align: left;">GO:0031994</td>
<td style="text-align: right;">0.012</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0005520</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">IGFBP2</td>
<td style="text-align: left;">insulin-like growth factor binding</td>
<td style="text-align: left;">GO:0005520</td>
<td style="text-align: right;">0.016</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0019838</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">IGFBP2</td>
<td style="text-align: left;">growth factor binding</td>
<td style="text-align: left;">GO:0019838</td>
<td style="text-align: right;">0.038</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0016765</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">MAT1A</td>
<td style="text-align: left;">transferase activity, transferring alkyl or aryl (other than methyl) groups</td>
<td style="text-align: left;">GO:0016765</td>
<td style="text-align: right;">0.025</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0033293</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">PC</td>
<td style="text-align: left;">monocarboxylic acid binding</td>
<td style="text-align: left;">GO:0033293</td>
<td style="text-align: right;">0.030</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0019842</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">PC</td>
<td style="text-align: left;">vitamin binding</td>
<td style="text-align: left;">GO:0019842</td>
<td style="text-align: right;">0.038</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0016874</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">PC</td>
<td style="text-align: left;">ligase activity</td>
<td style="text-align: left;">GO:0016874</td>
<td style="text-align: right;">0.040</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0031406</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">PC</td>
<td style="text-align: left;">carboxylic acid binding</td>
<td style="text-align: left;">GO:0031406</td>
<td style="text-align: right;">0.043</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0002020</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">SERPINA1</td>
<td style="text-align: left;">protease binding</td>
<td style="text-align: left;">GO:0002020</td>
<td style="text-align: right;">0.038</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0004867</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">SERPINA1/SERPINA3</td>
<td style="text-align: left;">serine-type endopeptidase inhibitor activity</td>
<td style="text-align: left;">GO:0004867</td>
<td style="text-align: right;">0.009</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0004866</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">SERPINA1/SERPINA3</td>
<td style="text-align: left;">endopeptidase inhibitor activity</td>
<td style="text-align: left;">GO:0004866</td>
<td style="text-align: right;">0.011</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0030414</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">SERPINA1/SERPINA3</td>
<td style="text-align: left;">peptidase inhibitor activity</td>
<td style="text-align: left;">GO:0030414</td>
<td style="text-align: right;">0.011</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0061135</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">SERPINA1/SERPINA3</td>
<td style="text-align: left;">endopeptidase regulator activity</td>
<td style="text-align: left;">GO:0061135</td>
<td style="text-align: right;">0.011</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0061134</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">SERPINA1/SERPINA3</td>
<td style="text-align: left;">peptidase regulator activity</td>
<td style="text-align: left;">GO:0061134</td>
<td style="text-align: right;">0.011</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0004857</td>
<td style="text-align: left;">Down</td>
<td style="text-align: left;">SERPINA1/SERPINA3</td>
<td style="text-align: left;">enzyme inhibitor activity</td>
<td style="text-align: left;">GO:0004857</td>
<td style="text-align: right;">0.017</td>
</tr>
</tbody>
</table>


</div>
</div>
</figure>
</div>
</div>
</section>
</section>
</section>
<section id="bibliography" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="bibliography">Bibliography</h2>
<div id="refs" class="references csl-bib-body" role="list" style="display: none">
<div id="ref-tidyverse" class="csl-entry" role="listitem">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">Wickham H, Averick M, Bryan J, et al (2019) Welcome to the <span></span>tidyverse<span></span>. 4:1686. <a href="https://doi.org/10.21105/joss.01686">https://doi.org/10.21105/joss.01686</a></div>
</div>
<div id="ref-DESeq2" class="csl-entry" role="listitem">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">Love MI, Huber W, Anders S (2014) Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. 15:550. <a href="https://doi.org/10.1186/s13059-014-0550-8">https://doi.org/10.1186/s13059-014-0550-8</a></div>
</div>
<div id="ref-org.Hs.eg.db" class="csl-entry" role="listitem">
<div class="csl-left-margin">3. </div><div class="csl-right-inline">Carlson M (2023) Org.hs.eg.db: Genome wide annotation for human</div>
</div>
<div id="ref-tibshirani1988" class="csl-entry" role="listitem">
<div class="csl-left-margin">4. </div><div class="csl-right-inline">Tibshirani R (1988) Estimating Transformations for Regression via Additivity and Variance Stabilization. Journal of the American Statistical Association 83(402):394–405. <a href="https://doi.org/10.1080/01621459.1988.10478610">https://doi.org/10.1080/01621459.1988.10478610</a></div>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./wat&amp;msld.html" class="pagination-link" aria-label="Adipose tissue and MASLD">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Adipose tissue and MASLD</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./wes.html" class="pagination-link" aria-label="Whole exome sequencing">
        <span class="nav-page-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Whole exome sequencing</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>